From d949a95e711742d645996457e7bb6f2bda6b8953 Mon Sep 17 00:00:00 2001
From: Roger WANG <roger.wang@intel.com>
Date: Tue, 22 Nov 2011 15:29:20 +0800
Subject: [PATCH 2/3] fix build errors in the previous commit

fix v8-gtk, build files for distribution
also fix code generator for higher version perl
split v8 binding from webcore to rebuild faster; linking with v8_g support
debian package files
more gitignore
suppress 'nullptr' build warning on Debian
make 'make dist' work
build hack
---
 .gitignore                                         |    7 +
 GNUmakefile.am                                     |   58 ++
 Source/JavaScriptCore/GNUmakefile.v8.am            |   16 +-
 Source/WebCore/GNUmakefile.am                      |   37 +-
 Source/WebCore/GNUmakefile.list.am                 |    8 +-
 .../bindings/scripts/CodeGeneratorGObject.pm       |    2 +-
 Source/WebCore/loader/icon/IconController.cpp      |    1 +
 Source/WebCore/platform/graphics/MediaPlayer.cpp   |    1 +
 Source/WebKit/gtk/GNUmakefile.am                   |    3 +-
 .../WebCoreSupport/FullscreenVideoController.cpp   |    1 +
 Source/WebKit/gtk/po/GNUmakefile.am                |    2 +-
 configure.ac                                       |   11 +-
 debian/changelog                                   |  785 +++++++++++++++++
 debian/compat                                      |    1 +
 debian/control                                     |   61 ++
 debian/copyright                                   |  923 ++++++++++++++++++++
 debian/libwebkitnode-1.0-0.install                 |    1 +
 debian/libwebkitnode-dev.install                   |    3 +
 debian/rules                                       |  160 ++++
 debian/source/format                               |    1 +
 debian/source/lintian-overrides                    |    1 +
 debian/watch                                       |    2 +
 22 files changed, 2074 insertions(+), 11 deletions(-)
 create mode 100644 debian/changelog
 create mode 100644 debian/compat
 create mode 100644 debian/control
 create mode 100644 debian/copyright
 create mode 100644 debian/libwebkitnode-1.0-0.install
 create mode 100644 debian/libwebkitnode-dev.install
 create mode 100644 debian/patches/series
 create mode 100755 debian/rules
 create mode 100644 debian/source/format
 create mode 100644 debian/source/lintian-overrides
 create mode 100644 debian/watch

diff --git a/.gitignore b/.gitignore
index 9c8649a..a75fc8a 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,7 +1,14 @@
+.deps/
+.libs/
+*.la
+*.mo
+*.lo
 *.mode*
 *.pbxuser
 *.perspective*
 *.pyc
+.dirstamp
+*.o
 .DS_Store
 build/
 /WebKitBuild/
diff --git a/GNUmakefile.am b/GNUmakefile.am
index bf6dac0..5a44229 100644
--- a/GNUmakefile.am
+++ b/GNUmakefile.am
@@ -272,3 +272,61 @@ clean-local:
 maintainer-clean-local: distclean-local
 distclean-local:
 	-rm -rf $(GENSOURCES) $(GENPROGRAMS)
+
+write_entries_to_file = echo $(DISTFILES) | tr ' ' '\n' > DISTFILES.list
+
+write_entries_to_file_split = 50
+write_entries_to_file_gnu = $(shell :>dist.lst) \
+   $(foreach range, \
+    $(shell i=1; while test $$i -le $(words $(DISTFILES)); do \
+      echo $$i; i=`expr $$i + $(write_entries_to_file_split)`; done), \
+    $(shell echo $(wordlist $(range), \
+     $(shell expr $(range) + $(write_entries_to_file_split) - 1), $(DISTFILES)) \
+      | tr ' ' '\n' >> dist.lst))
+
+$(eval write_entries_to_file = $$(write_entries_to_file_gnu))
+
+DISTFILES.list: GNUmakefile
+    $(write_entries_to_file)
+
+distdir: $(DISTFILES) dist.lst
+	$(am__remove_distdir)
+	test -d "$(distdir)" || mkdir "$(distdir)"
+	@srcdirstrip=`echo "$(srcdir)" | sed 's/[].[^$$\\*]/\\\\&/g'`; \
+	topsrcdirstrip=`echo "$(top_srcdir)" | sed 's/[].[^$$\\*]/\\\\&/g'`; \
+	  dist_files=`cat dist.lst | \
+	  sed -e "s|^$$srcdirstrip/||;t" \
+	      -e "s|^$$topsrcdirstrip/|$(top_builddir)/|;t"`; \
+	case $$dist_files in \
+	  */*) $(MKDIR_P) `echo "$$dist_files" | \
+			   sed '/\//!d;s|^|$(distdir)/|;s,/[^/]*$$,,' | \
+			   sort -u` ;; \
+	esac; \
+	for file in $$dist_files; do \
+	  if test -f $$file || test -d $$file; then d=.; else d=$(srcdir); fi; \
+	  if test -d $$d/$$file; then \
+	    dir=`echo "/$$file" | sed -e 's,/[^/]*$$,,'`; \
+	    if test -d "$(distdir)/$$file"; then \
+	      find "$(distdir)/$$file" -type d ! -perm -700 -exec chmod u+rwx {} \;; \
+	    fi; \
+	    if test -d $(srcdir)/$$file && test $$d != $(srcdir); then \
+	      cp -fpR $(srcdir)/$$file "$(distdir)$$dir" || exit 1; \
+	      find "$(distdir)/$$file" -type d ! -perm -700 -exec chmod u+rwx {} \;; \
+	    fi; \
+	    cp -fpR $$d/$$file "$(distdir)$$dir" || exit 1; \
+	  else \
+	    test -f "$(distdir)/$$file" \
+	    || cp -p $$d/$$file "$(distdir)/$$file" \
+	    || exit 1; \
+	  fi; \
+	done
+	$(MAKE) $(AM_MAKEFLAGS) \
+	  top_distdir="$(top_distdir)" distdir="$(distdir)" \
+	  dist-hook
+	-test -n "$(am__skip_mode_fix)" \
+	|| find "$(distdir)" -type d ! -perm -755 \
+		-exec chmod u+rwx,go+rx {} \; -o \
+	  ! -type d ! -perm -444 -links 1 -exec chmod a+r {} \; -o \
+	  ! -type d ! -perm -400 -exec chmod a+r {} \; -o \
+	  ! -type d ! -perm -444 -exec $(install_sh) -c -m a+r {} {} \; \
+	|| chmod -R a+r "$(distdir)"
diff --git a/Source/JavaScriptCore/GNUmakefile.v8.am b/Source/JavaScriptCore/GNUmakefile.v8.am
index 79b9371..b281264 100644
--- a/Source/JavaScriptCore/GNUmakefile.v8.am
+++ b/Source/JavaScriptCore/GNUmakefile.v8.am
@@ -10,6 +10,18 @@ javascriptcore_cppflags += \
 	-I$(srcdir)/Source/JavaScriptCore/runtime \
 	-I$(top_builddir)/Source/JavaScriptCore
 
+EXTRA_DIST += \
+	Source/JavaScriptCore/AUTHORS \
+	Source/JavaScriptCore/ChangeLog \
+	Source/JavaScriptCore/COPYING.LIB \
+	Source/JavaScriptCore/create_hash_table \
+	Source/JavaScriptCore/create_regex_tables \
+	Source/JavaScriptCore/icu/LICENSE \
+	Source/JavaScriptCore/icu/README \
+	Source/JavaScriptCore/KeywordLookupGenerator.py \
+	Source/JavaScriptCore/parser/Keywords.table \
+	Source/JavaScriptCore/THANKS
+
 javascriptcore_sources += \
 	Source/JavaScriptCore/wtf/Alignment.h \
 	Source/JavaScriptCore/wtf/AlwaysInline.h \
@@ -221,9 +233,7 @@ endif # USE_ICU_UNICODE
 # ----
 if USE_GLIB_UNICODE
 javascriptcore_sources += \
-	Source/JavaScriptCore/wtf/unicode/UnicodeMacrosFromICU.h \
-	Source/JavaScriptCore/wtf/unicode/gliUnicodeGLib.h \
-	Source/JavaScriptCore/wtf/unicode/gliUnicodeGLib.cpp
+	Source/JavaScriptCore/wtf/unicode/UnicodeMacrosFromICU.h
 endif # USE_GLIB_UNICODE
 
 Source/JavaScriptCore/RegExpJitTables.h: $(srcdir)/Source/JavaScriptCore/create_regex_tables
diff --git a/Source/WebCore/GNUmakefile.am b/Source/WebCore/GNUmakefile.am
index ac6dedd..db38f88 100644
--- a/Source/WebCore/GNUmakefile.am
+++ b/Source/WebCore/GNUmakefile.am
@@ -817,7 +817,7 @@ DerivedSources/WebCore/V8%.h: %.idl $(SCRIPTS_BINDINGS) $(WebCore)/bindings/scri
 endif # END USE_V8
 
 noinst_LTLIBRARIES += \
-	libWebCore.la
+	libWebCore.la libV8Binding.la
 
 nodist_EXTRA_libWebCore_la_SOURCES = \
 	$(webcore_built_nosources)
@@ -857,6 +857,38 @@ libWebCore_la_CPPFLAGS = \
 	$(XRENDER_CFLAGS) \
 	$(XT_CFLAGS)
 
+libV8Binding_la_SOURCES = \
+	$(v8binding_sources)
+
+libV8Binding_la_CXXFLAGS = \
+	$(SYMBOL_VISIBILITY_INLINES) \
+	$(global_cxxflags)
+
+libV8Binding_la_CFLAGS = \
+	$(SYMBOL_VISIBILITY) \
+	$(global_cflags)
+
+libV8Binding_la_CPPFLAGS = \
+	-DBUILDING_WEBKIT \
+	$(global_cppflags) \
+	$(webcore_cppflags) \
+	$(webcoregtk_cppflags) \
+	$(javascriptcore_cppflags) \
+	-fno-strict-aliasing \
+	$(CAIRO_CFLAGS) \
+	$(COVERAGE_CFLAGS) \
+	$(ENCHANT_CFLAGS) \
+	$(GEOCLUE_CFLAGS) \
+	$(GLIB_CFLAGS) \
+	$(GSTREAMER_CFLAGS) \
+	$(LIBSOUP_CFLAGS) \
+	$(LIBXML_CFLAGS) \
+	$(LIBXSLT_CFLAGS) \
+	$(SQLITE3_CFLAGS) \
+	$(UNICODE_CFLAGS) \
+	$(XRENDER_CFLAGS) \
+	$(XT_CFLAGS)
+
 # We have a different library with only the files that require GTK+. It allows us
 # to build a common WebCore lib and two different gtk WebCore libs depending on
 # the gtk version. In WeKit2 gtk2 is not supported, but the plugin process needs to
@@ -993,7 +1025,8 @@ endif # END USE_JSC
 
 if USE_V8
 EXTRA_DIST += \
-	Source/WebCore/bindings/scripts/CodeGeneratorV8.pm
+	Source/WebCore/bindings/scripts/CodeGeneratorV8.pm \
+        Source/WebCore/bindings/v8/custom/V8ArrayBufferViewCustomScript.js
 endif # END USE_V8
 
 # Installing web inspector files
diff --git a/Source/WebCore/GNUmakefile.list.am b/Source/WebCore/GNUmakefile.list.am
index 1d73119..56d85d9 100644
--- a/Source/WebCore/GNUmakefile.list.am
+++ b/Source/WebCore/GNUmakefile.list.am
@@ -3397,6 +3397,7 @@ webcore_sources += \
 	Source/WebCore/storage/IDBTransaction.h \
 	Source/WebCore/storage/IDBVersionChangeEvent.h \
 	Source/WebCore/storage/IDBVersionChangeRequest.h \
+        Source/WebCore/storage/IDBPendingTransactionMonitor.h \
 	Source/WebCore/storage/LocalStorageTask.cpp \
 	Source/WebCore/storage/LocalStorageTask.h \
 	Source/WebCore/storage/LocalStorageThread.cpp \
@@ -4066,6 +4067,11 @@ webcoregtk_sources += \
 	Source/WebCore/plugins/gtk/PluginPackageGtk.cpp \
 	Source/WebCore/plugins/gtk/PluginViewGtk.cpp
 
+if USE_V8
+webcoregtk_sources += \
+        Source/WebCore/platform/gtk/PlatformSupport.h
+endif
+
 if TARGET_X11
 webcoregtk_sources += \
 	Source/WebCore/platform/gtk/GtkWidgetBackingStoreX11.cpp \
@@ -5977,7 +5983,7 @@ endif # END USE_V8
 
 
 if USE_V8
-webcore_sources += \
+v8binding_sources = \
 	Source/WebCore/bindings/v8/DOMData.cpp \
 	Source/WebCore/bindings/v8/DOMData.h \
 	Source/WebCore/bindings/v8/DOMDataStore.cpp \
diff --git a/Source/WebCore/bindings/scripts/CodeGeneratorGObject.pm b/Source/WebCore/bindings/scripts/CodeGeneratorGObject.pm
index 9a02ca4..d7aa73b 100644
--- a/Source/WebCore/bindings/scripts/CodeGeneratorGObject.pm
+++ b/Source/WebCore/bindings/scripts/CodeGeneratorGObject.pm
@@ -1068,7 +1068,7 @@ sub GenerateFunctions {
         $param->name("value");
         $param->type($attribute->signature->type);
         my %attributes = ();
-        $param->extendedAttributes(attributes);
+        $param->extendedAttributes("attributes");
         my $arrayRef = $function->parameters;
         push(@$arrayRef, $param);
         
diff --git a/Source/WebCore/loader/icon/IconController.cpp b/Source/WebCore/loader/icon/IconController.cpp
index 2e5638c..00f9195 100644
--- a/Source/WebCore/loader/icon/IconController.cpp
+++ b/Source/WebCore/loader/icon/IconController.cpp
@@ -46,6 +46,7 @@
 #include "Logging.h"
 #include "Page.h"
 #include "Settings.h"
+#include "CString.h"
 
 namespace WebCore {
 
diff --git a/Source/WebCore/platform/graphics/MediaPlayer.cpp b/Source/WebCore/platform/graphics/MediaPlayer.cpp
index f97897d..9ddb99d 100644
--- a/Source/WebCore/platform/graphics/MediaPlayer.cpp
+++ b/Source/WebCore/platform/graphics/MediaPlayer.cpp
@@ -38,6 +38,7 @@
 #include "MediaPlayerPrivate.h"
 #include "Settings.h"
 #include "TimeRanges.h"
+#include "CString.h"
 
 #if PLATFORM(QT)
 #include <QtGlobal>
diff --git a/Source/WebKit/gtk/GNUmakefile.am b/Source/WebKit/gtk/GNUmakefile.am
index ccb7a62..7969160 100644
--- a/Source/WebKit/gtk/GNUmakefile.am
+++ b/Source/WebKit/gtk/GNUmakefile.am
@@ -88,7 +88,6 @@ libwebkitgtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LDFL
 libwebkitgtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LIBADD = \
 	-lpthread \
 	libWebCore.la \
-	libWebCoreGtk.la
 
 if USE_JSC
 libwebkitgtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LIBADD += \
@@ -96,6 +95,8 @@ libwebkitgtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LIBA
 endif # END USE_JSC
 
 libwebkitgtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LIBADD += \
+	libWebCoreGtk.la \
+        libV8Binding.la \
 	$(webcore_ldflags) \
 	$(CAIRO_LIBS) \
 	$(COVERAGE_LDFLAGS) \
diff --git a/Source/WebKit/gtk/WebCoreSupport/FullscreenVideoController.cpp b/Source/WebKit/gtk/WebCoreSupport/FullscreenVideoController.cpp
index 3300fe1..76a1c5c 100644
--- a/Source/WebKit/gtk/WebCoreSupport/FullscreenVideoController.cpp
+++ b/Source/WebKit/gtk/WebCoreSupport/FullscreenVideoController.cpp
@@ -26,6 +26,7 @@
 #include "GRefPtrGtk.h"
 #include "GtkVersioning.h"
 #include "MediaPlayer.h"
+#include "CString.h"
 
 #include <gdk/gdk.h>
 #include <gdk/gdkkeysyms.h>
diff --git a/Source/WebKit/gtk/po/GNUmakefile.am b/Source/WebKit/gtk/po/GNUmakefile.am
index 96e098d..3d0de58 100644
--- a/Source/WebKit/gtk/po/GNUmakefile.am
+++ b/Source/WebKit/gtk/po/GNUmakefile.am
@@ -15,7 +15,7 @@
 #
 # Origin: gettext-0.17
 
-DOMAIN += $(GETTEXT_PACKAGE)
+DOMAIN = $(GETTEXT_PACKAGE)
 
 MSGFMT := $(shell which msgfmt)
 
diff --git a/configure.ac b/configure.ac
index 657aca2..8c73260 100644
--- a/configure.ac
+++ b/configure.ac
@@ -103,6 +103,9 @@ fi
 # cleanly with that.
 if test "$CXX" = "g++"; then
    CXX_VERSION=`$CXX -dumpversion`
+   if test "$CXX_VERSION" = "4.6"; then
+       CXX_VERSION="4.6.0"
+   fi
    AX_COMPARE_VERSION([$CXX_VERSION],[ge],[4.6.0],CXXFLAGS="$CXXFLAGS -Wno-c++0x-compat")
 fi
 
@@ -862,8 +865,12 @@ AC_MSG_RESULT([$enable_jit])
 
 # Add V8 compiler flags and linker flags
 if test "$with_jsengine" = "v8"; then
-  V8_CPPFLAGS="-I\$(top_srcdir)/v8/include"
-  V8_LIBS="-L\$(top_srcdir)/v8/usr/lib -lv8"
+  if test "$enable_debug_symbols" = "yes"; then
+    V8_LIBS="-lv8"
+  else
+    V8_LIBS="-lv8"
+  fi
+
   AC_SUBST([V8_CPPFLAGS])
   AC_SUBST([V8_LIBS])
 fi
diff --git a/debian/changelog b/debian/changelog
new file mode 100644
index 0000000..8e1bd94
--- /dev/null
+++ b/debian/changelog
@@ -0,0 +1,785 @@
+webkit-node (1.7.0-1) unstable oneiric; urgency=low
+
+  * initial release based on debian package
+
+ -- Roger Wang <roger.wang@linux.intel.com>  Thu, 17 Nov 2011 20:27:29 +0800
+
+webkit (1.6.1-5) unstable; urgency=low
+
+  * debian/rules:
+  - also build with stabs debug symbols format when under s390x
+    patch by Aurelien Jarno <aurel32@debian.org> (Closes: #637228)
+  * debian/control:
+  - add Breaks for midori and claws-mail-fancy-plugin versions that
+    were using the private webkit_web_view_get_selected_text symbol
+    (Closes: #646095)
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Tue, 01 Nov 2011 23:20:19 -0200
+
+webkit (1.6.1-4) unstable; urgency=low
+
+  * debian/patches/03_freebsd-fix.patch:
+  - fix FreeBSD 64 bits build
+  * debian/patches/04_pkgconfig-fix.patch:
+  - import patch from upstream to fix pkg-config file to require the JSC
+    one
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Sun, 23 Oct 2011 20:10:53 -0200
+
+webkit (1.6.1-3) unstable; urgency=low
+
+  * debian/patches/01_double-conversion.patch:
+  - imported from upstream; fixes s390 double conversion, modified
+    to also include IA64
+  * debian/rules:
+  - try reducing the disk space used to build webkit by using thin
+    archives
+  - disable silent rules to make debugging easier
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Fri, 21 Oct 2011 10:38:17 -0200
+
+webkit (1.6.1-2) unstable; urgency=low
+
+  * Rename source package back to webkit
+  * debian/control:
+  - Build-Depend on gawk, since our GNUmakefile.in has lines that are too
+    long for mawk, which results in a broken GNUmakefile
+  * debian/rules:
+  - re-enable installing documentation
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Tue, 18 Oct 2011 09:59:20 -0200
+
+webkitgtk+ (1.6.1-1) experimental; urgency=low
+
+  * New upstream release
+  * debian/patches/01_fix-introspection.patch:
+  - Removed; applied upstream
+  * debian/patches/02_notebook_scroll.patch:
+  - Patch by Josselin Mouette <joss@debian.org> to ignore alt+wheel
+    (Closes: #640980)
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Mon, 26 Sep 2011 23:24:37 -0300
+
+webkitgtk+ (1.5.2-1) experimental; urgency=low
+
+  * New upstream development release
+  * debian/patches/01_spoof-ua-for-calendar.patch:
+  - Remove; applied upstream
+  * debian/patches/01_fix-introspection.patch:
+  - Fix the introspection build
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Wed, 24 Aug 2011 21:43:19 -0300
+
+webkitgtk+ (1.4.2-2) unstable; urgency=low
+
+  * debian/rules:
+  - fix armhf special-casing to actually work as intended; thanks to
+    mbiebl for finding the issue and helping resolve it
+  * debian/watch:
+  - provided by Michael Biebl <biebl@debian.org>; thanks! (Closes: #633446)
+  * debian/patches/01_spoof-ua-for-calendar.patch:
+  - steal patch from upstream (on track for 1.4.3) to make Google Calendar
+    behave and not give us the mobile version
+  * debian/control:
+  - Build-Depend on libjpeg-dev instead of libjpeg62-dev (See #635287)
+  - Re-add the libwebkit-dev package to automatically upgrade packages
+    using the GTK+ 2.0 version of WebKitGTK+.
+  - Fix capitalization of GTK+ in the description (Closes: #611871)
+  - Set Homepage to webkitgtk.org instead of webkit.org (Closes: #589431)
+  - Update to the new GI mini-policy by making the -dev packages depend on
+    the appropriate gir1.2-webkit-?.0 packages (Closes: #633962)
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Mon, 25 Jul 2011 21:08:52 -0300
+
+webkitgtk+ (1.4.2-1) unstable; urgency=low
+
+  * New upstream release
+  * debian/patches/01_add-missing-file.patch:
+  - removed; pushed upstream.
+  * debian/rules:
+  - use gstabs for mips and mipsel as well, to try to make it possible for
+    the builders to build webkit
+  - use -O1 for armhf to workaround compiler bug
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Sun, 10 Jul 2011 10:54:24 -0300
+
+webkitgtk+ (1.4.1-2) unstable; urgency=low
+
+  * debian/patches/01_add-missing-file.patch:
+  - copied MIPSAssembler.h from upstream's source repository; it's missing
+    from the sources list and didn't get included in the tarball, this will
+    fix the build on MIPS
+  * debian/control:
+  - fix dependency on gtk2 to please lintian (remove -1)
+  - adjust Standards-Version with no changes to the packaging
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Mon, 20 Jun 2011 14:50:27 -0300
+
+webkitgtk+ (1.4.1-1) unstable; urgency=low
+
+  * New upstream version, uploaded to unstable.
+  * debian/*.symbols:
+  - add optional symbols for some JIT stub symbols that show up in some
+    platforms (i.e. ARMEL)
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Mon, 06 Jun 2011 21:12:38 -0300
+
+webkitgtk+ (1.4.0-1) experimental; urgency=low
+
+  * debian/patches/02_plugin_check_mixed_symbols.patch,
+    debian/patches/01_plugin_path.patch:
+  - removed; applied upstream
+  * debian/control, debian/changelog:
+  - rename source package and remove the libwebkit-dev package to avoid
+    automatically transitioning packages to the new version
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Fri, 29 Apr 2011 10:57:23 -0300
+
+webkit (1.3.13-4) experimental; urgency=low
+
+  * debian/*.symbols:
+  - remove debian revision from WTFReportBacktrace@Base to make
+    lintian happy
+  - add optional C++ symbols for functions that use different data types
+    on 32 bits architectures
+  * debian/patches/02_plugin_check_mixed_symbols.patch:
+  - patch from upstream bugzilla that causes webkit itself to check for
+    mixed gtk symbols in a plugin before loading it
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Tue, 12 Apr 2011 09:21:29 -0300
+
+webkit (1.3.13-3) experimental; urgency=low
+
+  * debian/control:
+  - depend on a libgtk2.0-dev version that provides introspection information
+    (Closes: #621809)
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Fri, 08 Apr 2011 23:50:32 -0300
+
+webkit (1.3.13-2) experimental; urgency=low
+
+  * debian/patches/01_plugin_path.patch:
+  - obtained from upstream; will allow not disabling flash when used with
+    nspluginwrapper
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Fri, 08 Apr 2011 10:12:50 -0300
+
+webkit (1.3.13-1) experimental; urgency=low
+
+  * New upstream release.
+  * debian/patches/01_expose_less_symbols.patch:
+  - remove this patch; deal with the symbols instead
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Tue, 29 Mar 2011 11:14:00 -0300
+
+webkit (1.3.12-1) experimental; urgency=low
+
+  * New upstream release
+  * debian/*.symbols:
+  - updated with new symbols
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Tue, 15 Mar 2011 00:32:03 -0300
+
+webkit (1.3.11-3) experimental; urgency=low
+
+  * debian/*.symbols:
+  - fix demangled names (Closes: #614770)
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Wed, 23 Feb 2011 09:25:49 -0300
+
+webkit (1.3.11-2) experimental; urgency=low
+
+  * debian/libwebkit-{1,3}.0-0.symbols:
+  - tag exported WTF symbols as optional, and use their demangled name
+    with the c++ tag; this should fix the FTBFS
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Tue, 22 Feb 2011 14:31:14 -0300
+
+webkit (1.3.11-1) experimental; urgency=low
+
+  * New development release
+  * debian/patches/01_expose_less_symbols.patch:
+  - export less symbols, and don't build DRT
+  * debian/control:
+  - updated to reflect changes in naming of packages related to GTK+ 3
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Wed, 19 Jan 2011 11:05:31 -0200
+
+webkit (1.3.10-1) experimental; urgency=low
+
+  * New development release
+  * debian/control:
+  - Build-Depend on GTK+3 >= 2.99.1 to avoid problems with the multiple
+    breakages
+  * debian/rules:
+  - adapt for source layout changes
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Thu, 13 Jan 2011 10:37:55 -0200
+
+webkit (1.3.9-1) experimental; urgency=low
+
+  * New development release
+  - fixes build with newest GTK+
+  * debian/control:
+  - build-depend on GTK+ >= 2.91.7 for the GTK+3-based version
+  - build-depend on gir1.2-{gtk-2.0,soup-2.4}
+  - rename introspection packages to gir1.2-*
+  - updated Build-Depends on gobject-introspection
+  - tighten dependency of -dev packages on library packages
+  - add missing ${misc:Depends}
+  - update Standards-Version to 3.9.1
+  * debian/rules:
+  - add dh_girepository calls to actually do something to ${gir:Depends}
+  - run dh_makeshlibs targetting the appropriate packages (library ones,
+    not dev)
+  - remove rpath from the installed binaries
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Mon, 27 Dec 2010 14:06:08 -0200
+
+webkit (1.3.8-1) experimental; urgency=low
+
+  * New upstream development release
+  * debian/debian/gir1.0-webkit-1.0.install, debian/control,
+    debian/rules, debian/libwebkit-dev.install:
+  - add gir1.0-webkit-1.0 package, and enable introspection support
+  - rename packages to accomodate soname change
+  - add new packages for gtk3-based webkitgtk
+  * debian/control, debian/rules:
+  - enable building geolocation support
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Wed, 11 Aug 2010 11:10:28 -0300
+
+webkit (1.1.17-1) unstable; urgency=low
+
+  * New upstream release
+  * debian/copyright:
+  - updated with changes since 1.1.16
+  * debian/libwebkit-1.0-2.symbols:
+  - updated
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Wed, 02 Dec 2009 23:13:32 -0200
+
+webkit (1.1.16-3) unstable; urgency=low
+
+  * Fix building on kfresbsd-amd64 (Closes: #554718)
+    - Patch by Petr Salinger <Petr.Salinger@seznam.cz>
+  * Fix tightened dependency
+  * debian/rules:
+  - remove rpath from GtkLauncher and DumpRenderTree
+  * debian/control:
+  - depend on chrpath for change above
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Wed, 11 Nov 2009 21:21:12 -0200
+
+webkit (1.1.16-2) unstable; urgency=low
+
+  * debian/control:
+  - tighten libwebkit-dev dependency on the library
+    (Closes: #554014)
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Mon, 02 Nov 2009 17:07:03 -0200
+
+webkit (1.1.16-1) unstable; urgency=low
+
+  * New upstream release
+  * debian/copyright:
+  - updated with the copyright changes since 1.1.15
+  * debian/libwebkit-1.0-2.symbols:
+  - updated with new symbols
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Thu, 29 Oct 2009 19:06:20 -0200
+
+webkit (1.1.15.2-1) unstable; urgency=low
+
+  * New upstream (semi-stable) release
+  * Fix building on IA64 (Closes: #547797)
+    Thanks to Colin Watson <cjwatson@ubuntu.com>
+  * Fix building on alpha (Closes: #548499)
+  * Fix building on amd64 FreeBSD
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Tue, 06 Oct 2009 21:17:08 +0100
+
+webkit (1.1.15.1-1) unstable; urgency=low
+
+  * New upstream (semi-stable) release
+  * debian/libwebkit-1.0-2.symbols:
+  - remove revisions from the symbol versions
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Fri, 06 Aug 2010 11:19:52 -0300
+
+webkit (1.1.15-1) experimental; urgency=low
+
+  * New upstream release
+  - includes fix for crash that will be fixed in 1.1.15.1
+  * debian/copyright:
+  - updated with changes since 1.1.14
+  * debian/libwebkit-1.0-2.symbols:
+  - updated
+  * debian/control:
+  - force same upstream version for -common
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Mon, 21 Sep 2009 21:22:43 -0300
+
+webkit (1.1.14-1) experimental; urgency=low
+
+  * New upstream release
+  * debian/copyright:
+  - updated with changes since 1.1.13
+  * debian/control:
+  - made libwebkit-dev Arch: any (Closes: #539511)
+  * debian/libwebkit-1.0-2:
+  - updated
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Wed, 09 Sep 2009 11:32:23 -0300
+
+webkit (1.1.13-1) experimental; urgency=low
+
+  * New upstream release
+  - security fixes are already included in this release
+    (Closes: #538346, #538402)
+  * debian/control:
+  - update Build-Depends on libsoup to match upstream requirements
+    (Closes: #542272)
+  - Bump Build-Depends on libsoup2.4-dev to 2.27.91
+  * debian/copyright:
+  - updated with changes since 1.1.12
+  * debian/libwebkit-1.0-2.symbols:
+  - new symbols
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Mon, 24 Aug 2009 21:25:59 -0300
+
+webkit (1.1.12-1) unstable; urgency=low
+
+  * New upstream release
+  * debian/copyright:
+  - updated with changes since 1.1.10
+  * debian/libwebkit-1.0-2.symbols:
+  - adding new symbols
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Tue, 28 Jul 2009 21:04:56 +0200
+
+webkit (1.1.10-2) unstable; urgency=low
+
+  * debian/rules:
+  - trying to get the package built on s390, by using -gstabs
+    (Closes: #528524)
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Wed, 24 Jun 2009 23:05:07 -0300
+
+webkit (1.1.10-1) unstable; urgency=low
+
+  * debian/copyright:
+  - updated with the changes up to the current release
+  * debian/libwebkit-1.0-2.symbols:
+  - updated with new symbols
+  * debian/control:
+  - Build-Depend on libgail-dev
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Tue, 16 Jun 2009 18:52:32 -0300
+
+webkit (1.1.7-1) unstable; urgency=low
+
+  * New upstream release
+  * debian/libwebkit-1.0-2.symbols:
+  - updated with the new symbols in 1.1.7
+  * debian/libwebkit-dev.install, debian/libwebkit-dev.links,
+    debian/rules:
+  - Build, and ship gtk-doc documentation (Closes: #526683)
+  * debian/copyright:
+  - updated.
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Fri, 15 May 2009 18:30:58 -0300
+
+webkit (1.1.6-2) unstable; urgency=low
+
+  * Imported two changes from upstream to fix building on AMD64
+    (Closes: #527096)
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Tue, 12 May 2009 15:00:47 -0300
+
+webkit (1.1.6-1) unstable; urgency=low
+
+  [ Mike Hommey ]
+  * New upstream release.
+
+  [ Gustavo Noronha Silva ]
+  * Imported patch from upstream git mirror to avoid a regression in the
+    webkit_web_frame_load_string method.
+    See https://bugs.webkit.org/show_bug.cgi?id=25466
+  * debian/copyright:
+  - updated with changes since the last release
+  * debian/libwebkit-1.0-2.symbols:
+  - updated
+  * debian/control:
+  - added libenchant-dev to the Build-Depends
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Mon, 04 May 2009 21:40:15 -0300
+
+webkit (1.1.5-2) unstable; urgency=low
+
+  * Upload to unstable.
+  * debian/libwebkit-1.0-2.symbols:
+  - removed debian revision from the symbol versions
+  * debian/control:
+  - Put the libwebkit-1.0-2-dbg package in the "debug" section.
+  - Build depend on libgstreamer-plugins-base0.10-dev. (Closes: #525554)
+  * debian/rules:
+  - Bump shlibs to fit symbols file.
+
+ -- Mike Hommey <glandium@debian.org>  Sat, 25 Apr 2009 18:42:37 +0200
+
+webkit (1.1.5-1) experimental; urgency=low
+
+  [ Mike Hommey ]
+  * New upstream release:
+  - Filter out all C++ symbols. (Closes: #521569)
+  * debian/control:
+  - Change libwebkit-1.0-common section and priority to fit override.
+  - Bumped Standards-Version to 3.8.1.0. No changes required.
+  * debian/rules:
+  - Patch config.{sub|guess} in autotools/.
+
+  [ Gustavo Noronha Silva ]
+  * debian/copyright:
+  - updated with the copyright changes since last version
+  * debian/libwebkit-1.0-0.symbols:
+  - updated with the new symbols
+  * debian/libwebkit-1.0-common.install:
+  - also install translation files
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Thu, 16 Apr 2009 23:20:43 -0300
+
+webkit (1.1.3-3) experimental; urgency=low
+
+  [ Gustavo Noronha Silva ]
+  * debian/control:
+  - make libwebkit-1.0-dev depend on libsoup2.4-dev (>= 2.25.91)
+
+  [ Mike Hommey ]
+  * autotoolsconfig.h.in:
+  - Brown paper bag fix to add missing AC_DEFINEs.
+  * debian/control:
+  - libwebkit-dev doesn't need an outdated conflict with libwebkitgdk-dev.
+  - libwebkit-1.0-2-dbg doesn't need to conflict/replace libwebkit-1.0-1-dbg
+    thanks to the changes in previous release.
+  * debian/rules:
+  - Bump shlibs to fit symbols file.
+  * debian/copyright:
+  - Updated for a small discrepancy I spotted.
+  - Added Gustavo's copyright on debian/* files and updated mine.
+
+ -- Mike Hommey <glandium@debian.org>  Fri, 27 Mar 2009 21:20:48 +0100
+
+webkit (1.1.3-2) experimental; urgency=low
+
+  [ Gustavo Noronha Silva ]
+  * debian/control, debian/libwebkit-1.0-2.install:
+  - move the GtkLauncher and DumpRenderTree binaries to a directory named
+    after the soname of the library, so that different binary incompatible
+    versions of the library are installable in parallel (Closes: #520981)
+  * debian/libwebkit-1.0-2.symbols:
+  - removed debian revision from the symbol versions
+
+  [ Mike Hommey ]
+  * JavaScriptCore/jit/JITStubs.h:
+  - Use the first definition of RETURN_PAIR on x86-64.
+  * JavaScriptCore/assembler/MacroAssemblerX86_64.h,
+    JavaScriptCore/GNUmakefile.am, GNUmakefile.in, configure.ac, configure:
+  - Enable JIT on x86-64.
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Thu, 26 Mar 2009 17:46:30 -0300
+
+webkit (1.1.3-1) experimental; urgency=low
+
+  [ Gustavo Noronha Silva ]
+  * New upstream release
+  - correctly scrolls with space bar (Closes: #519883)
+  * debian/copyright:
+  - updated with changes since last release
+  * debian/libwebkit-1.0-2.symbols:
+  - added new symbols
+
+  [ Mike Hommey ]
+  * autotools/symbols.filter, debian/libwebkit-1.0-2.symbols:
+  - Hide JIT implementation specific symbols. (Closes: #519924, #520006)
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Tue, 17 Mar 2009 15:05:40 -0300
+
+webkit (1.1.1-1) experimental; urgency=low
+
+  * New 1.1.1 upstream release
+  * debian/rules, debian/control, debian/*:
+  - soversion was bumped, so rename packages accordingly
+  * debian/control:
+  - curl is not used anymore, so drop build-dependecy and replace it with
+    libsoup2.4 >= 2.25.91
+  * debian/libwebkit-1.0-2.symbols:
+  - updated symbols file; bumped required version of all symbols to 1.1.1,
+    since we had a soversion bump
+  * debian/copyright:
+  - updated with changes from last release we shipped, and updated
+    download location
+  * debian/control, debian/libwebkit-1.0-common.install:
+  - new package to distribute data files, such as the Web Inspector files
+  - libwebkit-1.0-2 now depends on libwebkit-1.0-common
+  * debian/control:
+  - add myself as an Uploader
+
+ -- Gustavo Noronha Silva <kov@debian.org>  Thu, 05 Mar 2009 16:28:43 -0300
+
+webkit (1.0.2~pre.svn37878-1) experimental; urgency=low
+
+  * Upstream 1.0.2 pre-release, taken from svn revision 37878.
+  * debian/rules:
+    - Use configure instead of autogen.sh, now configure is shipped.
+    - Don't remove automake/autoconf generated files.
+    - Bump shlibs because of some new symbols.
+    - Don't need --enable-svg-experimental anymore.
+    - Feed configure with --host and --build.
+    - Use config.guess and config.sub from autotools-dev.
+  * debian/control: Don't build depend on autoconf/automake/libtool, but
+    autotools-dev, instead.
+  * GNUmakefile.am, GNUmakefile.in: Add missing dash in version script option.
+  * debian/libwebkit-1.0-1.symbols: Add new symbols.
+  * WebCore/page/gtk/AXObjectCacheAtk.cpp: printfs being removed upstream,
+    we don't need to #include <stdio.h> anymore.
+  * configure.ac, configure, GNUmakefile.am, GNUmakefile.in: Properly check
+    and link against libpng. bz#21884.
+  * configure.ac, configure: Properly check and link against libpangoft2.
+    bz#21885.
+  * debian/copyright: Updated to fit additions/removals of files upstream.
+
+ -- Mike Hommey <glandium@debian.org>  Sat, 25 Oct 2008 15:19:56 +0200
+
+webkit (1.0.1-4) unstable; urgency=high
+
+  * WebCore/dom/Document.*, WebCore/loader/DocLoader.*: Avoid DoS via
+    crafted CSS import statements. Fixes: CVE-2008-3632. Closes: #499771.
+
+ -- Mike Hommey <glandium@debian.org>  Sat, 27 Sep 2008 08:57:48 +0200
+
+webkit (1.0.1-3) unstable; urgency=low
+
+  * WebCore/platform/graphics/gtk/FontCacheGtk.cpp,
+    WebCore/platform/graphics/gtk/FontGtk.cpp,
+    WebCore/platform/graphics/gtk/FontPlatformData.h,
+    WebCore/platform/graphics/gtk/FontPlatformDataGtk.cpp,
+    WebCore/platform/graphics/gtk/SimpleFontDataGtk.cpp: cherry-picked change
+    from SVN revision 36309 to fix various font selection problems.
+    Closes: #464477.
+
+ -- Mike Hommey <glandium@debian.org>  Sat, 20 Sep 2008 12:02:58 +0200
+
+webkit (1.0.1-2) unstable; urgency=low
+
+  * symbols.filter: As a workaround for #490173, hide all C++ mangled symbols.
+    This will be enough for now, while fixing FTBFS on ARM.
+  * debian/rules: Build with -Wl,--no-relax on alpha, to work around a
+    binutils bug causing FTBFS.
+
+ -- Mike Hommey <glandium@debian.org>  Thu, 10 Jul 2008 21:10:27 +0200
+
+webkit (1.0.1-1) unstable; urgency=low
+
+  * New upstream release. Closes: #489385.
+  * debian/copyright:
+    - Updated to fit additions/removals of files upstream.
+    - Updated where the source was gotten.
+    - Fixed typos for Collabora. Closes: #484661.
+  * JavaScriptCore/wtf/FastMalloc.cpp, JavaScriptCore/wtf/ListHashSet.h,
+    JavaScriptCore/wtf/Platform.h, JavaScriptCore/wtf/Vector.h,
+    WebCore/platform/text/AtomicString.cpp,
+    WebCore/platform/text/StringHash.h: Fixed some alignment problems on sparc
+    (and some that might occur on arm, too). Closes: #487745. Some compiler
+    warnings about alignment remain, but I don't know if they are a real
+    problem yet.
+  * debian/control:
+    - Added build dependency on libxt-dev.
+    - Relax libwebkit-dev dependency on libwebkit-1.0-1.
+    - Bumped Standards-Version to 3.8.0.1. No changes.
+  * WebCore/page/gtk/AXObjectCacheAtk.cpp: Include stdio.h to avoid FTBFS
+    because of undefined printf.
+  * GNUmakefile.am, symbols.filter: Filter out all std::* symbols exported
+    because of stl headers.
+  * debian/libwebkit-1.0-1.symbols: Add symbols file.
+  * debian/rules: Bump shlibs because of some new symbols.
+
+ -- Mike Hommey <glandium@debian.org>  Sun, 06 Jul 2008 15:09:55 +0200
+
+webkit (0~svn32442-1) unstable; urgency=low
+
+  [ Mike Hommey ]
+  * New upstream snapshot
+  * debian/copyright: Updated to fit additions/removals of files upstream.
+  * debian/control: Add libpango1.0-dev to build dependencies and tighten
+    libgtk2.0-dev build dependency. Closes: #477493.
+
+  [ Luca Bruno ]
+  * debian/libwebkit-1.0-1.install, debian/rules: Install GtkLauncher
+    and DumpRenderTree in /usr/lib/webkit-1.0/libexec. Closes: #476514.
+
+ -- Mike Hommey <glandium@debian.org>  Wed, 23 Apr 2008 22:20:25 +0200
+
+webkit (0~svn31841-1) unstable; urgency=low
+
+  * New upstream snapshot
+    + Includes proper Requires in .pc file. Closes: #450949.
+  * debian/copyright: Updated to fit additions/removals of files upstream.
+  * debian/control, debian/rules: Don't build QtWebKit, it will be built from
+    Qt sources.
+  * debian/libqtwebkit*: Removed.
+  * debian/rules: Replaced make calls with $(MAKE).
+  * debian/control, debian/rules: Don't use qmake anymore, but use autotools
+    instead.
+  * debian/libwebkitgtk-dev.install: Remove the .prl file that was generated
+    by qmake.
+  * WebCore/WebCore.pro, WebKit.pro: Revert previous changes, as we don't use
+    qmake anymore.
+  * JavaScriptCore/wtf/ASCIICType.h: Revert what were warning fixes, to avoid
+    diverging too much from upstream.
+  * debian/rules: Clean-up autotools generated files in the clean target.
+  * debian/control, debian/rules, debian/libwebkit*: Rename packages to fit
+    new library SONAME.
+  * debian/control: libwebkit-dev doesn't need to depend on libglib2.0-dev.
+  * debian/libwebkit-1.0-1.install, debian/rules: Install GtkLauncher and
+    DumpRenderTree in /usr/lib/webkit-1.0, and don't rename DumpRenderTree, as
+    Qt homonym lie elsewhere.
+  * debian/libwebkit-1.0-1.preinst, debian/libwebkit-1.0-1.postrm: Don't
+    divert anything, since we don't clash with older files anymore.
+  * debian/rules: Ugly hack to add -Wl,--as-needed without it getting reordered
+    by libtool.
+  * WebKitTools/GNUmakefile.am: Don't build GtkLauncher and DumpRenderTree
+    with rpath.
+
+ -- Mike Hommey <glandium@debian.org>  Sat, 12 Apr 2008 21:40:00 +0200
+
+webkit (0~svn27674-4) unstable; urgency=low
+
+  * debian/rules: Revert change from 0~svn27674-3, which was obviously not
+    enough.
+  * WebKit.pro: Don't build testkjs, that provokes the s390 FTBFSes ; we don't
+    install it anyways. Closes: #466848.
+  * JavaScriptCore/kjs/interpreter.cpp, JavaScriptCore/wtf/ASCIICType.h,
+    WebCore/dom/Position.cpp, WebCore/editing/Selection.cpp,
+    WebCore/editing/SelectionController.cpp,
+    WebCore/editing/VisiblePosition.cpp,
+    WebCore/ksvg2/svg/SVGFontFaceElement.cpp,
+    WebCore/loader/FTPDirectoryParser.cpp,
+    WebCore/loader/icon/IconDatabase.cpp,
+    WebCore/platform/TextCodecLatin1.cpp,
+    WebCore/platform/TextCodecUserDefined.cpp,
+    WebCore/platform/TextStream.cpp,
+    WebCore/platform/image-decoders/jpeg/JPEGImageDecoder.cpp: Fix FTBFS with
+    gcc 4.3 due to missing includes. Closes: #455125.
+
+ -- Mike Hommey <glandium@debian.org>  Thu, 21 Feb 2008 23:01:12 +0100
+
+webkit (0~svn27674-3) unstable; urgency=low
+
+  * debian/rules: Work around #466613 (FTBFS on s390) by building a little bit
+    less optimized (-O1). Thanks to Josselin Mouette.
+  * debian/control:
+    - Add Vcs-Browser and Vcs-Git fields, and turn Homepage indications in
+      descriptions into a control field.
+    - Bumped Standards-Version to 3.7.3.0. No changes.
+
+ -- Mike Hommey <glandium@debian.org>  Wed, 20 Feb 2008 21:49:33 +0100
+
+webkit (0~svn29752-1) experimental; urgency=low
+
+  * New upstream snapshot
+  * debian/copyright: Updated to fit additions/removals of files upstream.
+  * JavaScriptCore/wtf/TCSpinLock.h: Revert our work-around, now that a
+    proper patch has been applied upstream.
+  * WebCore/WebCore.pro: Don't use Qt version as SO version for QtWebKit.
+  * debian/control, debian/rules, debian/lib*0d.install: Bump SO version to
+    1d because of ABI incompatible changes, and change package names
+    accordingly.
+  * debian/rules: Don't remove -lqtwebico from QtWebKit.pc, since it's not
+    here anymore.
+  * debian/rules, debian/lib*1d.install:
+    - Install new Gtk port's DumpRenderTree tool.
+    - Rename both port's DumpRenderTree tools to <port name>DumpRenderTree
+      to avoid conflicting names.
+  * debian/lib*1d.postrm, debian/lib*1d.preinst: Avoid conflicting files with
+    lib*0d packages (*Launcher programs) but allow to install both new and old
+    libraries by using diversions.
+
+ -- Mike Hommey <glandium@debian.org>  Fri, 25 Jan 2008 00:31:51 +0100
+
+webkit (0~svn27674-2) unstable; urgency=low
+
+  * JavaScriptCore/JavaScriptCore.pri: cherry-picked change from revision
+    28692 to fix FTBFS due to lack of -lpthread on the linker command line.
+  * debian/control: Add dependencies on necessary development packages
+    (essentially for header files) to our own development packages.
+
+ -- Mike Hommey <glandium@debian.org>  Sat, 15 Dec 2007 11:04:47 +0100
+
+webkit (0~svn27674-1) unstable; urgency=low
+
+  * New upstream snapshot
+  * debian/copyright: Updated to fit additions/removals of files upstream.
+  * debian/control: Make libwebkitgtk-dev conflict with the old
+    libwebkitgdk-dev. Closes: #449001.
+  * debian/rules: Bump qtwebkit shlibs.
+  * WebKit/qt/Api/qwebpage.cpp: cherry-picked change from revision 27904 to
+    fix crashes when an event is caught outside of the webkit frame in Qt.
+    This occurred, for example, when hovering over the QtLauncher toolbar.
+
+ -- Mike Hommey <glandium@debian.org>  Mon, 19 Nov 2007 23:10:38 +0100
+
+webkit (0~svn26044-1) unstable; urgency=low
+
+  * New upstream snapshot
+  * debian/rules:
+    - Add support for DEB_BUILD_OPTIONS=noopt.
+    - Bump qtwebkit shlibs, and remove versioning on the webkitgtk ones,
+      as the library is new.
+  * debian/copyright: Updated to fit additions/removals of files upstream.
+  * debian/control, debian/rules, debian/libwebkitgtk-dev.install,
+    debian/libwebkitgtk0d.install: Replace occurences of gdk by gtk, and
+    rename libwebkitgdk*, to fit upstream rename of the Gtk port.
+    Closes: #445060.
+
+ -- Mike Hommey <glandium@debian.org>  Fri, 05 Oct 2007 00:05:06 +0200
+
+webkit (0~svn25144-2) unstable; urgency=low
+
+  * JavaScriptCore/wtf/Platform.h:
+    - Also test if __arm__ is defined, which should fix the FTBFS on arm.
+    - Use better defines for our various arm ports.
+  * JavaScriptCore/kjs/ustring.h, WebCore/platform/DeprecatedString.h: Use
+    these new defines. Thanks Riku Voipio.
+  * debian/control: Build depend on Qt >= 4.3. Thanks Hubert Figuiere.
+    Closes: #439672.
+  * debian/rules: Explicitely use qmake-qt4 instead of qmake to avoid build
+    failures when qt3-dev-tools is installed. Thanks Michael Biebl.
+    Closes: #441007.
+
+ -- Mike Hommey <glandium@debian.org>  Thu, 06 Sep 2007 08:12:21 +0200
+
+webkit (0~svn25144-1) unstable; urgency=low
+
+  * New upstream snapshot
+  * debian/copyright: Updated so as to fit what we actually remove (there were
+    missing removals previously, which were not appropriate for the most
+    anyways), and to fit the additions/removals of files upstream.
+  * JavaScriptCore/wtf/TCSpinLock.h: Work around an FTBFS on PPC due to a
+    probable regression in gcc (#438415).
+  * debian/rules:
+    + Change the place we install QtLauncher from, since it moved.
+    + Set binary packages' shlibs correctly.
+    + Use $(CURDIR) variable more safely to avoid problem with build
+      directories with spaces.
+  * WebKitQt/Plugins/Plugins.pro: Build plugins with hidden symbols, so that
+    they don't expose unwanted symbols.
+
+ -- Mike Hommey <glandium@debian.org>  Sun, 19 Aug 2007 15:54:12 +0200
+
+webkit (0~svn24735-1) unstable; urgency=low
+
+  * Initial release. (Closes: #428855)
+
+ -- Mike Hommey <glandium@debian.org>  Wed, 15 Aug 2007 14:19:46 +0200
+
diff --git a/debian/compat b/debian/compat
new file mode 100644
index 0000000..7ed6ff8
--- /dev/null
+++ b/debian/compat
@@ -0,0 +1 @@
+5
diff --git a/debian/control b/debian/control
new file mode 100644
index 0000000..59ba82f
--- /dev/null
+++ b/debian/control
@@ -0,0 +1,61 @@
+Source: webkit-node
+Priority: optional
+Section: web
+Maintainer: Roger Wang <roger.wang@linux.intel.com>
+Build-Depends: gawk, chrpath, debhelper (>= 5.0), libgtk2.0-dev (>= 2.24.3), gir1.2-gtk-2.0, libgtk-3-dev, gir1.2-gtk-3.0 (>= 3.0.0), libpango1.0-dev (>= 1.16), libicu-dev, libxslt-dev, libsoup2.4-dev (>= 2.32.2), gir1.2-soup-2.4, libsqlite3-dev, gperf, bison, flex, libjpeg-dev, libpng12-dev, libxt-dev, autotools-dev, libgstreamer-plugins-base0.10-dev, libenchant-dev, gtk-doc-tools, libgail-dev, libgail-3-dev, libgeoclue-dev, gobject-introspection (>= 0.9.12-4), libgirepository1.0-dev (>= 0.9.12-4), libv8-dev
+Build-Conflicts: gir-repository-dev
+Standards-Version: 3.9.2
+Homepage: http://webkitgtk.org/
+Vcs-Browser: http://git.debian.org/?p=pkg-webkit/webkit.git
+Vcs-Git: git://git.debian.org/git/pkg-webkit/webkit.git
+
+Package: libwebkitnode-1.0-0
+Section: libs
+Architecture: any
+Depends: ${shlibs:Depends},
+         ${misc:Depends}
+Breaks: midori (<< 0.4.1-2), claws-mail-fancy-plugin (<< 3.7.10-3)
+Description: Web content engine library for node-webkit
+ WebKit is a web content engine, derived from KHTML and KJS from KDE, and
+ used primarily in Apple's Safari browser.  It is made to be embedded in
+ other applications, such as mail readers, or web browsers.
+ .
+ It is able to display content such as HTML, SVG, XML, and others. It also
+ supports DOM, XMLHttpRequest, XSLT, CSS, Javascript/ECMAscript and more.
+ .
+ This is the library for node-webkit.
+
+Package: libwebkitnode-dev
+Section: libdevel
+Priority: extra
+Architecture: any
+Depends: libwebkitnode-1.0-0 (= ${binary:Version}),
+         libgtk2.0-dev (>= 2.24.3),
+         libsoup2.4-dev (>= 2.25.91),
+         ${misc:Depends}
+Description: Web content engine library for node-webkit - Development files
+ WebKit is a web content engine, derived from KHTML and KJS from KDE, and
+ used primarily in Apple's Safari browser.  It is made to be embedded in
+ other applications, such as mail readers, or web browsers.
+ .
+ It is able to display content such as HTML, SVG, XML, and others. It also
+ supports DOM, XMLHttpRequest, XSLT, CSS, Javascript/ECMAscript and more.
+ .
+ This package provides development files required to build node-webkit.
+
+Package: libwebkitnode-1.0-0-dbg
+Section: debug
+Priority: extra
+Architecture: any
+Depends: libwebkitnode-1.0-0 (= ${binary:Version}),
+         ${misc:Depends}
+Description: Web content engine library for node-webkit - Debugging symbols
+ WebKit is a web content engine, derived from KHTML and KJS from KDE, and
+ used primarily in Apple's Safari browser.  It is made to be embedded in
+ other applications, such as mail readers, or web browsers.
+ .
+ It is able to display content such as HTML, SVG, XML, and others. It also
+ supports DOM, XMLHttpRequest, XSLT, CSS, Javascript/ECMAscript and more.
+ .
+ This package provides the debugging symbols for the Webkit library for node-webkit.
+
diff --git a/debian/copyright b/debian/copyright
new file mode 100644
index 0000000..5a6c79f
--- /dev/null
+++ b/debian/copyright
@@ -0,0 +1,923 @@
+Original source may be found at http://webkitgtk.org/
+
+Files: debian/*
+Copyright: © 2007-2009 Mike Hommey <glandium@debian.org>
+           © 2009 Gustavo Noronha Silva <kov@debian.org>
+License: LGPL-2+ (/usr/share/common-licenses/LGPL-2)
+
+Files: JavaScriptCore/API/*,
+       JavaScriptCore/assembler/*,
+       JavaScriptCore/VM/CTI.*,
+       JavaScriptCore/profiler/*,
+       JavaScriptCore/wtf/Assertions.*,
+       JavaScriptCore/wtf/FastAllocBase.h,
+       JavaScriptCore/wtf/HashIterators.h,
+       JavaScriptCore/wtf/MathExtras.h,
+       JavaScriptCore/wtf/Platform.h,
+       JavaScriptCore/wtf/StringExtras.h,
+       JavaScriptCore/wtf/unicode/UTF8.*,
+       JavaScriptCore/yarr/*,
+       WebCore/bindings/js/DOMObjectWithSVGContext.h,
+       WebCore/bindings/js/JSAudioConstructor.*,
+       WebCore/bindings/js/JSConsoleCustom.cpp,
+       WebCore/bindings/js/JSCoordinatesCustom.cpp,
+       WebCore/bindings/js/JSCSSRuleCustom.cpp,
+       WebCore/bindings/js/JSCSSStyleDeclarationCustom.*,
+       WebCore/bindings/js/JSCSSValueCustom.cpp,
+       WebCore/bindings/js/JSCustomXPathNSResolver.*,
+       WebCore/bindings/js/JSDOMApplicationCacheCustom.cpp,
+       WebCore/bindings/js/JSEventTargetBase.*,
+       WebCore/bindings/js/JSHTMLAllCollection.h,
+       WebCore/bindings/js/JSHTMLAppletElementCustom.*,
+       WebCore/bindings/js/JSHTMLDocumentCustom.cpp,
+       WebCore/bindings/js/JSHTMLElementCustom.cpp,
+       WebCore/bindings/js/JSHTMLEmbedElementCustom.*,
+       WebCore/bindings/js/JSHTMLFormElementCustom.cpp,
+       WebCore/bindings/js/JSHTMLFrameSetElementCustom.cpp,
+       WebCore/bindings/js/JSHTMLObjectElementCustom.*,
+       WebCore/bindings/js/JSHistoryCustom.h,
+       WebCore/bindings/js/JSInspectedObjectWrapper.*,
+       WebCore/bindings/js/JSInspectorCallbackWrapper.*,
+       WebCore/bindings/js/JSJavaScriptCallFrameCustom.cpp,
+       WebCore/bindings/js/JSLocationCustom.h,
+       WebCore/bindings/js/JSMessagePortCustom.*,
+       WebCore/bindings/js/JSNamedNodeMapCustom.cpp,
+       WebCore/bindings/js/JSNamedNodesCollection.*,
+       WebCore/bindings/js/JSNodeCustom.cpp,
+       WebCore/bindings/js/JSNodeFilterCustom.cpp,
+       WebCore/bindings/js/JSNodeListCustom.cpp,
+       WebCore/bindings/js/JSQuarantinedObjectWrapper.*,
+       WebCore/bindings/js/JSSVGPODTypeWrapper.h,
+       WebCore/bindings/js/JSStorageCustom.*,
+       WebCore/bindings/js/JSStyleSheetCustom.cpp,
+       WebCore/bindings/js/JSStyleSheetListCustom.cpp,
+       WebCore/bindings/js/JSTextCustom.cpp,
+       WebCore/bindings/js/JSXMLHttpRequestUploadCustom.*,
+       WebCore/bindings/js/JSXSLTProcessorConstructor.*,
+       WebCore/bindings/js/JSWorkerContextCustom.*,
+       WebCore/bindings/js/JSWorkerCustom.*,
+       WebCore/bindings/js/GCController.*,
+       WebCore/bindings/js/ScriptSourceProvider.h,
+       WebCore/bridge/*,
+       WebCore/css/CSSCanvasValue.*,
+       WebCore/css/CSSFontFace.*,
+       WebCore/css/CSSFontFaceSource.*,
+       WebCore/css/CSSFontFaceSrcValue.*,
+       WebCore/css/CSSFontSelector.*,
+       WebCore/css/CSSGradientValue.*,
+       WebCore/css/CSSImageGeneratorValue.*,
+       WebCore/css/CSSPrimitiveValueMappings.h,
+       WebCore/css/CSSReflectValue.*,
+       WebCore/css/CSSReflectionDirection.h,
+       WebCore/css/CSSRuleList.idl,
+       WebCore/css/CSSSegmentedFontFace.*,
+       WebCore/css/CSSTimingFunctionValue.*,
+       WebCore/css/CSSUnicodeRangeValue.*,
+       WebCore/css/CSSValueList.idl,
+       WebCore/css/MediaList.idl,
+       WebCore/css/MediaQuery*,
+       WebCore/css/view-source.css,
+       WebCore/css/svg.css,
+       WebCore/dom/BeforeTextInsertedEvent.*,
+       WebCore/dom/Clipboard.cpp,
+       WebCore/dom/ClipboardAccessPolicy.h,
+       WebCore/dom/EventTarget.cpp,
+       WebCore/dom/EventTarget.h,
+       WebCore/dom/ExceptionCode.cpp,
+       WebCore/dom/MessageEvent.*,
+       WebCore/dom/NodeWithIndex.h,
+       WebCore/dom/OverflowEvent.*,
+       WebCore/dom/Position*,
+       WebCore/dom/ProgressEvent.*,
+       WebCore/dom/RangeBoundaryPoint.h,
+       WebCore/dom/DOMStringList.h,
+       WebCore/dom/TextEvent.*,
+       WebCore/editing/*,
+       WebCore/history/*,
+       WebCore/html/Canvas*,
+       WebCore/html/File*,
+       WebCore/html/HTMLAudioElement.*,
+       WebCore/html/HTMLCanvasElement.*,
+       WebCore/html/HTMLMediaElement.*,
+       WebCore/html/HTMLParserErrorCodes.*,
+       WebCore/html/HTMLSourceElement.*,
+       WebCore/html/HTMLVideoElement.*,
+       WebCore/html/HTMLViewSourceDocument.*,
+       WebCore/html/MediaError.h,
+       WebCore/html/PreloadScanner.*,
+       WebCore/html/TimeRanges.*,
+       WebCore/html/VoidCallback.h,
+       WebCore/inspector/*,
+       WebCore/loader/CachePolicy.h,
+       WebCore/loader/CachedFont.*,
+       WebCore/loader/ImageDocument.*,
+       WebCore/loader/PluginDocument.*,
+       WebCore/loader/ProgressTracker.*,
+       WebCore/loader/SubstituteData.h,
+       WebCore/loader/SubstituteResource.h,
+       WebCore/loader/TextDocument.*,
+       WebCore/loader/FTPDirectoryDocument.*,
+       WebCore/loader/*/*,
+       WebCore/page/*,
+       WebCore/platform/*,
+       WebCore/plugins/gtk/PluginPackageGtk.cpp,
+       WebCore/plugins/gtk/PluginViewGtk.cpp,
+       WebCore/plugins/npapi.cpp,
+       WebCore/plugins/npfunctions.h,
+       WebCore/plugins/PluginDatabase.*,
+       WebCore/plugins/PluginDatabaseClient.h,
+       WebCore/plugins/PluginDebug.h,
+       WebCore/plugins/PluginInfoStore.*,
+       WebCore/plugins/PluginStream.*,
+       WebCore/plugins/PluginPackage.*,
+       WebCore/plugins/PluginView.*,
+       WebCore/plugins/PluginQuirkSet.h,
+       WebCore/rendering/LayoutState.*,
+       WebCore/rendering/RenderHTMLCanvas.*,
+       WebCore/rendering/RenderImageGeneratedContent.*,
+       WebCore/rendering/RenderMedia.*,
+       WebCore/rendering/RenderTreeAsText.*,
+       WebCore/rendering/RenderWordBreak.*,
+       WebCore/rendering/SVGRenderTreeAsText.*,
+       WebCore/rendering/RenderVideo.*,
+       WebCore/storage/LocalStorage.*,
+       WebCore/storage/LocalStorageArea.*,
+       WebCore/storage/LocalStorageTask.*,
+       WebCore/storage/LocalStorageThread.*,
+       WebCore/storage/SessionStorage.*,
+       WebCore/storage/SessionStorageArea.*,
+       WebCore/storage/Storage.*,
+       WebCore/storage/StorageArea.*,
+       WebCore/storage/StorageEvent.*,
+       WebCore/storage/StorageMap.*,
+       WebCore/svg/ElementTimeControl.h,
+       WebCore/svg/*.idl,
+       WebCore/svg/animation/*,
+       WebCore/svg/graphics/SVGImage.*,
+       WebCore/svg/graphics/SVGPaintServer*,
+       WebCore/svg/graphics/SVGResource*,
+       WebCore/xml/*,
+       WebCore/workers/Worker.cpp,
+       WebCore/workers/Worker.h,
+       WebCore/workers/WorkerContext.cpp,
+       WebCore/workers/WorkerContext.h,
+       WebCore/workers/WorkerImportScriptsClient.*,
+       WebCore/workers/WorkerLocation.cpp,
+       WebCore/workers/WorkerLocation.h,
+       WebCore/workers/WorkerMessagingProxy.cpp,
+       WebCore/workers/WorkerMessagingProxy.h,
+       WebCore/workers/WorkerThread.cpp,
+       WebCore/workers/WorkerThread.h,
+       WebKit/gtk/WebCoreSupport/ContextMenuClientGtk.h,
+       WebKit/gtk/WebCoreSupport/DocumentLoaderGtk.h,
+       WebKit/gtk/WebCoreSupport/EditorClientGtk.h,
+       WebKit/gtk/WebCoreSupport/FrameLoaderClientGtk.h,
+       WebKitTools/DumpRenderTree/AccessibilityController.h,
+       WebKitTools/DumpRenderTree/AccessibilityUIElement.h,
+       WebKitTools/DumpRenderTree/gtk/EventSender.*,
+       WebKitTools/GtkLauncher/main.c,
+Copyright: © 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009 Apple Inc.
+           © 2005, 2006, 2007 Alexey Proskuryakov (ap@nypop.com)
+           © 2006, 2007, 2008 Nikolas Zimmermann <zimmermann@kde.org>
+           © 2006 Samuel Weinig <sam.weinig@gmail.com>
+           © 2005 Frerich Raabe <raabe@kde.org>
+           © 2005 Maksim Orlovich <maksim@kde.org>
+           © 2005, 2006 Alexander Kellett <lypanov@kde.org>
+           © 2005, 2006 Kimmo Kinnunen <kimmo.t.kinnunen@nokia.com>
+           © 2005, 2006 Michael Emmel mike.emmel@gmail.com
+           © 2005, 2009 Google Inc.
+           © 2005, 2006, 2009 Nokia Corporation
+           © 2005, 2006 Oliver Hunt <ojh16@student.canterbury.ac.nz>
+           © 2006, 2007, 2008 Eric Seidel <eric@webkit.org>
+           © 2006 Allan Sandfeld Jensen <sandfeld@kde.org>
+           © 2006 David Smith (catfish.man@gmail.com)
+           © 2006 Dirk Mueller <mueller@kde.org>
+           © 2006 Justin Haygood <jhaygood@spsu.edu>
+           © 2006, 2007 Rob Buis <buis@kde.org>
+           © 2006 Simon Hausmann <hausmann@kde.org>
+           © 2006 Zack Rusin <zack@kde.org>
+           © 2006 Graham Dennis (graham.dennis@gmail.com)
+           © 2007, 2008 Alp Toker <alp.toker@collabora.co.uk>
+           © 2007 Holger Hans Peter Freyther <zecke@selfish.org>
+           © 2007 Henry Mason (hmason@mac.com)
+           © 2007 Hiroyuki Ikezoe
+           © 2007 Kouhei Sutou
+           © 2007 Christian Dywan <christian@twotoasts.de>
+           © 2007 Marius Renn <damarvy@gmail.com>
+           © 2008 Xan Lopez <xan@gnome.org>
+           © 2008 Collabora, Ltd.
+           © 2008 Julien Chaffraix <jchaffraix@webkit.org>
+           © 2008 The Mozilla Foundation
+           © 2008 Brent Fulgham <bfulgham@gmail.com>
+           © 2008 Collin Jackson  <collinj@webkit.org>
+           © 2008, 2009 Dirk Schulze <krit@webkit.org>
+           © 2008, 2009 Torch Mobile Inc.
+           © 2008 Gustavo Noronha Silva <gns@gnome.org>
+           © 2008 Kelvin W Sherlock (ksherlock@gmail.com)
+           © 2008 Matt Lilek <webkit@mattlilek.com>
+           © 2008, 2009 Paul Pedriana <ppedriana@ea.com>
+           © 2009 Daniel Bates (dbates@intudata.com)
+           © 2009 Joseph Pecoraro
+           © 2009 IBM Corp.
+           © 2009 Anthony Ricaud <rik@webkit.org>
+License: BSD-2
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions
+ are met:
+ 1. Redistributions of source code must retain the above copyright
+    notice, this list of conditions and the following disclaimer.
+ 2. Redistributions in binary form must reproduce the above copyright
+    notice, this list of conditions and the following disclaimer in the
+    documentation and/or other materials provided with the distribution.
+ .
+ THIS SOFTWARE IS PROVIDED BY APPLE COMPUTER, INC. ``AS IS'' AND ANY
+ EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE COMPUTER, INC. OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+Files: JavaScriptCore/API/JSRetainPtr.h,
+       JavaScriptCore/API/JSStringRefBSTR.h,
+       JavaScriptCore/VM/*,
+       JavaScriptCore/kjs/CallData.h,
+       JavaScriptCore/kjs/ConstructData.h,
+       JavaScriptCore/kjs/DebuggerCallFrame.*,
+       JavaScriptCore/kjs/InitializeThreading.*,
+       JavaScriptCore/kjs/JSActivation.*,
+       JavaScriptCore/kjs/JSGlobalData.*,
+       JavaScriptCore/kjs/JSGlobalObject.cpp,
+       JavaScriptCore/kjs/JSNotAnObject.*,
+       JavaScriptCore/kjs/JSVariableObject.*,
+       JavaScriptCore/kjs/SourceProvider.h,
+       JavaScriptCore/kjs/SourceRange.h,
+       JavaScriptCore/kjs/SymbolTable.h,
+       JavaScriptCore/profiler/ProfileNode.*,
+       JavaScriptCore/profiler/Profiler.*,
+       JavaScriptCore/wtf/ASCIICType.h,
+       JavaScriptCore/wtf/AVLTree.h,
+       JavaScriptCore/wtf/Deque.h,
+       JavaScriptCore/wtf/DisallowCType.h,
+       JavaScriptCore/wtf/Locker.h,
+       JavaScriptCore/wtf/MainThread.*,
+       JavaScriptCore/wtf/MallocZoneSupport.h,
+       JavaScriptCore/wtf/MessageQueue.h,
+       JavaScriptCore/wtf/Threading.h,
+       JavaScriptCore/wtf/ThreadingPthreads.cpp,
+       JavaScriptCore/wtf/ThreadSpecific.h,
+       JavaScriptCore/wtf/gtk/MainThreadGtk.cpp,
+       JavaScriptCore/wtf/gtk/ThreadingGtk.cpp,
+       JavaScriptCore/wtf/unicode/Collator.h,
+       JavaScriptCore/wtf/unicode/CollatorDefault.cpp,
+       JavaScriptCore/wtf/unicode/icu/CollatorICU.cpp,
+       JavaScriptCore/wtf/*,
+       WebCore/accessibility/Accessibility*,
+       WebCore/accessibility/AXObjectCache.cpp,
+       WebCore/bindings/js/JSAttrCustom.cpp,
+       WebCore/bindings/js/JSClipboardCustom.cpp,
+       WebCore/bindings/js/JSCustomSQL*,
+       WebCore/bindings/js/JSCustomVoidCallback.*,
+       WebCore/bindings/js/JSDatabaseCustom.cpp,
+       WebCore/bindings/js/JSHistoryCustom.cpp,
+       WebCore/bindings/js/JSEventCustom.cpp,
+       WebCore/bindings/js/JSHTMLFrameElementCustom.cpp,
+       WebCore/bindings/js/JSHTMLIFrameElementCustom.cpp,
+       WebCore/bindings/js/JSHTMLSelectElementCustom.h,
+       WebCore/bindings/js/JSSQLResultSetRowListCustom.cpp,
+       WebCore/bindings/js/JSSQLTransactionCustom.cpp,
+       WebCore/bindings/js/JSXMLHttpRequestCustom.cpp,
+       WebCore/bindings/js/JSXSLTProcessorCustom.cpp,
+       WebCore/bindings/js/StringSourceProvider.h,
+       WebCore/bridge/npruntime.h,
+       WebCore/dom/ClassNodeList.*,
+       WebCore/dom/Clipboard.idl,
+       WebCore/dom/DOMCoreException.idl,
+       WebCore/dom/DOMCoreException.h,
+       WebCore/dom/EventException.idl,
+       WebCore/dom/EventException.h,
+       WebCore/dom/ExceptionBase.*,
+       WebCore/dom/SelectorNodeList.*,
+       WebCore/dom/StaticNodeList.*,
+       WebCore/dom/make_names.pl,
+       WebCore/editing/SmartReplace*,
+       WebCore/history/CachedPagePlatformData.h,
+       WebCore/html/CanvasPixelArray.*,
+       WebCore/html/HTMLTableRowsCollection.*,
+       WebCore/html/ImageData.*,
+       WebCore/loader/archive/*,
+       WebCore/loader/DocumentLoader.*,
+       WebCore/loader/FormState.*,
+       WebCore/loader/FrameLoader*,
+       WebCore/loader/MainResourceLoader.*,
+       WebCore/loader/NavigationAction.*,
+       WebCore/loader/NetscapePlugInStreamLoader.*,
+       WebCore/loader/ResourceLoader.*,
+       WebCore/loader/SubresourceLoader*,
+       WebCore/loader/icon/IconDatabaseClient.h,
+       WebCore/loader/icon/IconRecord.*,
+       WebCore/loader/icon/PageURLRecord.*,
+       WebCore/page/animation/*,
+       WebCore/page/BarInfo.*,
+       WebCore/page/Console.*,
+       WebCore/page/DOMSelection.*,
+       WebCore/page/Location.*,
+       WebCore/page/Screen.*,
+       WebCore/page/SecurityOrigin*,
+       WebCore/page/WindowFeatures.h,
+       WebCore/platform/AutodrainedPool.h,
+       WebCore/platform/FileChooser.*,
+       WebCore/platform/FileSystem.h,
+       WebCore/platform/FloatConversion.h,
+       WebCore/platform/graphics/FontCache.*,
+       WebCore/platform/graphics/FontFallbackList.cpp,
+       WebCore/platform/graphics/FontDescription.cpp,
+       WebCore/platform/graphics/Glyph*,
+       WebCore/platform/graphics/SimpleFontData.cpp,
+       WebCore/platform/graphics/StringTruncator.*,
+       WebCore/platform/graphics/gtk/GlyphPageTreeNodeGtk.cpp,
+       WebCore/platform/graphics/gtk/GlyphPageTreeNodePango.cpp,
+       WebCore/platform/graphics/gtk/IconGtk.cpp,
+       WebCore/platform/graphics/gtk/SimpleFontDataGtk.cpp,
+       WebCore/platform/graphics/gtk/SimpleFontDataPango.cpp,
+       WebCore/platform/gtk/ContextMenuGtk.cpp,
+       WebCore/platform/gtk/ContextMenuItemGtk.cpp,
+       WebCore/platform/gtk/PasteboardGtk.cpp,
+       WebCore/platform/gtk/SearchPopupMenuGtk.cpp,
+       WebCore/platform/network/HTTPParsers.*,
+       WebCore/platform/sql/SQLValue.*,
+       WebCore/platform/sql/SQLiteAuthorizer.*,
+       WebCore/platform/text/AtomicStringHash.h,
+       WebCore/platform/text/StringBuffer.h,
+       WebCore/platform/text/StringBuilder.*,
+       WebCore/platform/text/mac/make-charset-table.pl,
+       WebCore/plugins/PluginQuirkSet.h,
+       WebCore/rendering/MediaControlElements.*,
+       WebCore/rendering/RenderListBox.*,
+       WebCore/rendering/RenderReplica.*,
+       WebCore/storage/*,
+       WebCore/xml/XMLHttpRequest.idl,
+       WebCore/xml/XMLHttpRequestException.h,
+       WebCore/xml/XMLHttpRequestException.idl,
+       WebCore/xml/XPathException.h,
+       WebCore/xml/XPathException.idl,
+       WebCore/xml/XSLTUnicodeSort.*,
+       WebKitTools/DumpRenderTree/*,
+       WebKit/*,
+       WebKit/gtk/*,
+Copyright: © 2003, 2005, 2006, 2007, 2008, 2009 Apple Inc.
+           © 2004, Apple Computer, Inc. and The Mozilla Foundation
+           © 2006 Samuel Weinig <sam.weinig@gmail.com>
+           © 2006 Alexey Proskuryakov (ap@nypop.com)
+           © 2007, 2008 Matt Lilek (pewtermoose@gmail.com)
+           © 2006 Michael Emmel mike.emmel@gmail.com
+           © 2007, 2008 Alp Toker <alp@atoker.com>
+           © 2006, 2007 David Smith (catfish.man@gmail.com)
+           © 2006 Nikolas Zimmermann <zimmermann@kde.org>
+           © 2006 Eric Seidel (eric@webkit.org)
+           © 2006 Graham Dennis (graham.dennis@gmail.com)
+           © 2007 Justin Haygood (jhaygood@reaktix.com)
+           © 2008 Cameron Zwarich (cwzwarich@uwaterloo.ca)
+           © 2008 Tony Chang <idealisms@gmail.com>
+           © 2008 Anthony Ricaud (rik24d@gmail.com)
+           © 2008, 2009 Google Inc.
+           © 2008, 2009 Torch Mobile Inc.
+           © 2009 Jian Li <jianli@chromium.org>
+License: BSD-3
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions
+ are met:
+ .
+ 1.  Redistributions of source code must retain the above copyright
+     notice, this list of conditions and the following disclaimer.
+ 2.  Redistributions in binary form must reproduce the above copyright
+     notice, this list of conditions and the following disclaimer in the
+     documentation and/or other materials provided with the distribution.
+ 3.  Neither the name of Apple, Inc. ("Apple") nor the names of
+     its contributors may be used to endorse or promote products derived
+     from this software without specific prior written permission.
+ .
+ THIS SOFTWARE IS PROVIDED BY APPLE AND ITS CONTRIBUTORS "AS IS" AND ANY
+ EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL APPLE OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+ DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+Files: WebCore/inspector/front-end/WatchExpressionsSidebarPane.js
+License: BSD-3
+Copyright: © 2009 IBM Corp. All rights reserved.
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions are
+ met:
+
+     * Redistributions of source code must retain the above copyright
+ notice, this list of conditions and the following disclaimer.
+     * Redistributions in binary form must reproduce the above
+ copyright notice, this list of conditions and the following disclaimer
+ in the documentation and/or other materials provided with the
+ distribution.
+     * Neither the name of IBM Corp. nor the names of its
+ contributors may be used to endorse or promote products derived from
+ this software without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+
+Files: JavaScriptCore/wtf/HashTable.*,
+       JavaScriptCore/wtf/unicode/glib/*,
+       JavaScriptCore/runtime/*,
+       WebCore/accessibility/gtk/AXObjectCacheAtk.cpp,
+       WebCore/accessibility/gtk/AccessibilityObjectAtk.cpp,
+       WebCore/accessibility/gtk/AccessibilityObjectWrapperAtk.*,
+       WebCore/bindings/js/JSEventListener.*,
+       WebCore/bindings/js/JSLazyEventListener.*,
+       WebCore/bindings/v8/custom/V8DocumentLocationCustom.cpp
+       WebCore/css/makegrammar.pl,
+       WebCore/dom/XMLTokenizer.*,
+       WebCore/editing/gtk/SelectionControllerGtk.cpp,
+       WebCore/html/FormDataList.*,
+       WebCore/html/HTMLAnchorElement.*,
+       WebCore/html/HTMLAppletElement.*,
+       WebCore/html/HTMLAreaElement.*,
+       WebCore/html/HTMLCollection.*,
+       WebCore/html/HTMLFormCollection.*,
+       WebCore/html/HTMLFrameElementBase.*,
+       WebCore/html/HTMLIFrameElement.*,
+       WebCore/html/HTMLFontElement.*,
+       WebCore/html/HTMLFormElement.*,
+       WebCore/html/HTMLInputElement.*,
+       WebCore/html/HTMLLinkElement.*,
+       WebCore/html/HTMLObjectElement.*,
+       WebCore/html/HTMLOptionsCollection.*,
+       WebCore/html/HTMLPlugInImageElement.*,
+       WebCore/html/HTMLTableElement.*,
+       WebCore/html/HTMLTextAreaElement.*,
+       WebCore/html/HTMLTokenizer.*,
+       WebCore/html/ValidityState.*,
+       WebCore/page/Chrome.*,
+       WebCore/page/ChromeClient.h,
+       WebCore/page/Frame.*,
+       WebCore/page/FramePrivate.h,
+       WebCore/page/FrameTree.*,
+       WebCore/page/FrameView.*,
+       WebCore/page/MouseEventWithHitTestResults.*,
+       WebCore/page/Navigator.*,
+       WebCore/page/Page.*,
+       WebCore/page/PrintContext.*,
+       WebCore/page/WindowFeatures.cpp,
+       WebCore/platform/Length*,
+       WebCore/platform/PopupMenuClient.h,
+       WebCore/platform/PopupMenu.h,
+       WebCore/platform/SearchPopupMenu.h,
+       WebCore/platform/StaticConstructors.h,
+       WebCore/platform/TreeShared.h,
+       WebCore/platform/graphics/filters/*,
+       WebCore/platform/graphics/FloatPoint3D.*,
+       WebCore/platform/graphics/Font.cpp,
+       WebCore/platform/graphics/Font.h,
+       WebCore/platform/graphics/FontDescription.h,
+       WebCore/platform/graphics/FontFallbackList.h,
+       WebCore/platform/graphics/Icon.h,
+       WebCore/platform/graphics/IntSizeHash.h,
+       WebCore/platform/graphics/PathTraversalState.cpp,
+       WebCore/platform/graphics/SimpleFontData.h,
+       WebCore/platform/graphics/TextRun.h,
+       WebCore/platform/graphics/WidthIterator.*,
+       WebCore/platform/graphics/cairo/CairoPath.h,
+       WebCore/platform/graphics/cairo/PathCairo.cpp,
+       WebCore/platform/graphics/gtk/FontCacheGtk.cpp,
+       WebCore/platform/graphics/gtk/FontGtk.*,
+       WebCore/platform/graphics/gtk/FontPlatformData.h,
+       WebCore/platform/graphics/gtk/FontPlatformDataGtk.cpp,
+       WebCore/platform/graphics/gtk/FontPlatformDataPango.cpp,
+       WebCore/platform/graphics/gtk/MediaPlayerPrivateGStreamer.*,
+       WebCore/platform/graphics/gtk/VideoSinkGStreamer.*,
+       WebCore/platform/graphics/transforms/*,
+       WebCore/platform/gtk/ClipboardGtk.cpp,
+       WebCore/platform/gtk/ContextMenuGtk.cpp,
+       WebCore/platform/gtk/ContextMenuItemGtk.cpp,
+       WebCore/platform/gtk/CursorGtk.h,
+       WebCore/platform/gtk/DragDataGtk.cpp,
+       WebCore/platform/gtk/DragImageGtk.cpp,
+       WebCore/platform/gtk/EventLoopGtk.cpp,
+       WebCore/platform/gtk/FileSystemGtk.cpp,
+       WebCore/platform/gtk/guriescape.c,
+       WebCore/platform/gtk/guriescape.h,
+       WebCore/platform/gtk/KURLGtk.cpp,
+       WebCore/platform/gtk/Language.cpp,
+       WebCore/platform/gtk/LoggingGtk.cpp,
+       WebCore/platform/gtk/PasteboardGtk.cpp,
+       WebCore/platform/gtk/PasteboardHelper.h,
+       WebCore/platform/gtk/PopupMenuGtk.cpp,
+       WebCore/platform/gtk/RenderThemeGtk.cpp,
+       WebCore/platform/gtk/RenderThemeGtk.h,
+       WebCore/platform/gtk/ScrollbarGtk.cpp,
+       WebCore/platform/gtk/SearchPopupMenuGtk.cpp,
+       WebCore/platform/gtk/SharedBufferGtk.cpp,
+       WebCore/platform/gtk/SoundGtk.cpp,
+       WebCore/platform/gtk/SystemTimeGtk.cpp,
+       WebCore/platform/network/FormData.*,
+       WebCore/platform/network/curl/CookieJarCurl.cpp,
+       WebCore/platform/network/soup/CookieJarSoup.cpp,
+       WebCore/platform/network/soup/ResourceHandleSoup.cpp,
+       WebCore/platform/network/soup/ResourceRequestSoup.cpp,
+       WebCore/platform/network/soup/ResourceResponseSoup.cpp,
+       WebCore/platform/text/AtomicString.*,
+       WebCore/platform/text/AtomicStringImpl.h,
+       WebCore/platform/text/Base64.cpp,
+       WebCore/platform/text/BidiContext.*,
+       WebCore/platform/text/BidiResolver.h,
+       WebCore/platform/text/ParserUtilities.h,
+       WebCore/platform/text/PlatformString.h,
+       WebCore/platform/text/String.cpp,
+       WebCore/platform/text/SegmentedString.*,
+       WebCore/platform/text/StringHash.h,
+       WebCore/platform/text/StringImpl.*,
+       WebCore/platform/text/TextBreakIterator.h,
+       WebCore/platform/text/TextBreakIteratorICU.cpp,
+       WebCore/platform/text/TextBreakIteratorInternalICU.h,
+       WebCore/platform/text/gtk/TextBreakIteratorInternalICUGtk.cpp,
+       WebCore/rendering/ScrollBehavior.*,
+       WebCore/svg/SVGAngle.idl,
+       WebCore/svg/SVGColor.idl,
+       WebCore/svg/SVGDocument.idl,
+       WebCore/svg/SVGElement.idl,
+       WebCore/svg/SVGException.idl,
+       WebCore/svg/SVGFontData.*,
+       WebCore/svg/SVGHKernElement.idl,
+       WebCore/svg/SVGLength.idl,
+       WebCore/svg/SVGMatrix.idl,
+       WebCore/svg/SVGMetadataElement.idl,
+       WebCore/svg/SVGNumber.idl,
+       WebCore/svg/SVGPoint.idl,
+       WebCore/svg/SVGRect.idl,
+       WebCore/svg/SVGSVGElement.idl,
+       WebCore/svg/SVGTransform.idl,
+       WebCore/svg/graphics/SVGResourceFilter.*,
+       WebCore/svg/graphics/filters/*,
+       WebCore/xml/DOMParser.*,
+       WebCore/xml/XMLHttpRequest.*,
+       WebCore/xml/XMLSerializer.*,
+       WebCore/xml/XPathEvaluator.idl,
+       WebCore/xml/XPathExpression.idl,
+       WebCore/xml/XPathNSResolver.idl,
+       WebCore/xml/XPathResult.idl,
+       WebCore/xml/XSL*,
+       WebCore/wml/*,
+       WebKit/gtk/WebCoreSupport/*.cpp,
+       WebKit/gtk/WebCoreSupport/ChromeClientGtk.h,
+       WebKit/gtk/WebCoreSupport/PasteboardHelperGtk.h,
+       WebKit/gtk/WebCoreSupport/InspectorClientGtk.h,
+       WebKit/gtk/webkit/*,
+       WebKit/gtk/tests/*,
+       WebKitTools/DumpRenderTree/config.h,
+       WebKitTools/DumpRenderTree/gtk/WorkQueueItemGtk.cpp,
+       *
+Copyright: © 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009 Apple Inc.
+           © 1991, 1999 Free Software Foundation, Inc.
+           © 1997 Martin Jones (mjones@kde.org)
+           © 1998, 1999 Torben Weis <weis@kde.org>
+           © 1998 Netscape Communications Corporation
+           © 1998-2003, 2006 Lars Knoll (knoll@kde.org)
+           © 1999-2004 Harri Porten (porten@kde.org)
+           © 1999, 2001 Antti Koivisto (koivisto@kde.org)
+           © 1999, 2002 Waldo Bastian (bastian@kde.org)
+           © 2000 Daniel Molkentin (molkentin@kde.org)
+           © 2000-2003 Dirk Mueller (mueller@kde.org)
+           © 2000 Frederik Holljen (frederik.holljen@hig.no)
+           © 2000, 2001, 2003 Peter Kelly (pmk@post.com)
+           © 2000 Simon Hausmann <hausmann@kde.org>
+           © 2000 Stefan Schimanski (schimmi@kde.org)
+           © 2001 Tobias Anton (anton@stud.fbi.fh-darmstadt.de)
+           © 2001 Tim Copperfield <timecop@network.email.ne.jp>
+           © 2002, 2003 The Karbon Developers
+           © 2002 Cyrus Patel <cyp@fb14.uni-mainz.de>
+           © 2004, 2005, 2006, 2007 Nikolas Zimmermann <zimmermann@kde.org>
+           © 2004, 2005, 2006, 2007, 2008, 2009 Rob Buis <buis@kde.org>
+           © 2004, 2005 Allan Sandfeld Jensen (kde@carewolf.com)
+           © 2004, 2006 Zack Rusin <zack@kde.org>
+           © 2005, 2006 Alexander Kellett <lypanov@kde.org>
+           © 2005, 2006, 2007 Alexey Proskuryakov <ap@nypop.com>
+           © 2005, 2006 Oliver Hunt <ojh16@student.canterbury.ac.nz>
+           © 2006, 2007 Eric Seidel (eric@webkit.org)
+           © 2006, 2007 Samuel Weinig <sam@webkit.org>
+           © 2006 Anders Carlsson <andersca@mac.com>
+           © 2006 Andrew Wellington (proton@wiretapped.net)
+           © 2006 Bjoern Graf (bjoern.graf@gmail.com)
+           © 2006 George Staikos <staikos@kde.org>
+           © 2006 Graham Dennis (graham.dennis@gmail.com)
+           © 2006 James G. Speth (speth@end.com)
+           © 2006 Jon Shier (jshier@iastate.edu)
+           © 2006, 2007 Maks Orlovich
+           © 2006 Michael Emmel <mike.emmel@gmail.com>
+           © 2007, 2008 Alp Toker <alp@atoker.com>
+           © 2007 Cameron Zwarich (cwzwarich@uwaterloo.ca)
+           © 2007, 2008, 2009 Holger Hans Peter Freyther <zecke@selfish.org>
+           © 2007 Krzysztof Kowalczyk <kkowalczyk@gmail.com>
+           © 2007 David Smith (catfish.man@gmail.com)
+           © 2007, 2008, 2009 Collabora, Ltd.
+           © 2007 OpenedHand
+           © 2007, 2008, 2009 Christian Dywan <christian@twotoasts.de>
+           © 2007,2008 Luca Bruno <lethalman88@gmail.com>
+           © 2008 Xan Lopez <xan@gnome.org>
+           © 2008 Nuanti Ltd
+           © 2008 Nokia Corporation and/or its subsidiary(-ies)
+           © 2007, 2008 Christian Dywan <christian@imiendo.com>
+           © 2008, 2009 Torch Mobile, Inc. All rights reserved.
+           © 2008 Alex Mathews <possessedpenguinbob@gmail.com>
+           © 2008 David Levin <levin@chromium.org>
+           © 2009 David Smith (catfish.man@gmail.com)
+           © 2008, 2009 Gustavo Noronha Silva <gns@gnome.org>
+           © 2008, 2009 Jan Michael Alonzo <jmalonzo@gmail.com>
+           © 2008 Jürg Billeter <j@bitron.ch>
+           © 2008 Dominik Röttsches <dominik.roettsches@access-company.com>
+           © 2009 Igalia S.L.
+           © 2009 Google Inc.
+           © 2009 Diego Escalante Urrelo <diegoe@gnome.org>
+           © 2009 Michelangelo De Simone <micdesim@gmail.com>
+           © 2009 Kenneth Rohde Christiansen
+           © 2009 Cameron McCormack <cam@mcc.id.au>
+           © 2009 Martin Robinson
+License: LGPL-2+ (/usr/share/common-licenses/LGPL-2)
+
+Files: WebCore/bridge/npruntime_internal.h
+Copyright: © 2007 Collabora, Ltd.
+License: LGPL-2.1+ (/usr/share/common-licenses/LGPL-2)
+
+Files: JavaScriptCore/kjs/DateMath.*,
+       WebCore/bridge/npapi.h,
+       WebCore/html/HTMLDocument.cpp,
+       WebCore/platform/Arena.*,
+       WebCore/platform/gtk/gtk2drawing.c,
+       WebCore/platform/gtk/gtkdrawing.h,
+       WebCore/platform/image-decoders/gif/GIFImageReader.*,
+       WebCore/platform/image-decoders/jpeg/JPEGImageDecoder.cpp,
+       WebCore/platform/image-decoders/png/PNGImageDecoder.cpp,
+       WebCore/platform/image-decoders/zlib/mozzconf.h,
+       WebCore/platform/text/UnicodeRange.*,
+       WebCore/plugins/gtk/gtk2xtbin.*,
+       WebCore/plugins/gtk/xembed.h,
+       WebCore/rendering/RenderArena.*,
+       WebCore/rendering/RenderLayer.*
+Copyright: © 1998, 2001-2006 mozilla.org
+           © 1999, 2000 Harri Porten (porten@kde.org)
+           © 2003, 2004, 2005, 2006, 2007 Apple Inc.
+           © 1998-2000, 2002 Netscape Communications Corporation.
+           © 1999 Lars Knoll (knoll@kde.org)
+           © 1999 Antti Koivisto (koivisto@kde.org)
+           © 2004 IBM Corporation
+           © 2002 Sun Microsystems, Inc.
+License: LGPL-2.1+ (/usr/share/common-licenses/LGPL-2) | GPL-2+ (/usr/share/common-licenses/GPL-2) | MPL-1.1
+ Other contributors for this code include:
+ - Boris Zbarsky <bzbarsky@mit.edu>
+ - Brian Ryner <bryner@brianryner.com>
+ - Chris Saari <saari@netscape.com>
+ - Christian Biesinger <cbiesinger@web.de>
+ - David Baron <dbaron@fas.harvard.edu>
+ - Jeff Hostetler <jeff@nerdone.com>
+ - Josh Soref <timeless@mac.com>
+ - Michael Ventnor <m.ventnor@gmail.com>
+ - Nick Blievers <nickb@adacel.com.au>
+ - Pierre Chanial <p_ch@verizon.net>
+ - Raffaele Sena <raff@netwinder.org>
+ - Randall Jesup <rjesup@wgate.com>
+ - Robert O'Callahan <roc+@cs.cmu.edu>
+ - Roland Mainz <roland.mainz@informatik.med.uni-giessen.de>
+ - Stuart Parmenter <stuart@mozilla.com>
+ - Tom Rini <trini@kernel.crashing.org>
+
+Files: JavaScriptCore/pcre/*
+Copyright: © 1997-2005 University of Cambridge
+           © 2002, 2004, 2005, 2006, 2007 Apple Inc.
+License: BSD-3
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions are met:
+ .
+     * Redistributions of source code must retain the above copyright notice,
+       this list of conditions and the following disclaimer.
+ .
+     * Redistributions in binary form must reproduce the above copyright
+       notice, this list of conditions and the following disclaimer in the
+       documentation and/or other materials provided with the distribution.
+ .
+     * Neither the name of the University of Cambridge nor the names of its
+       contributors may be used to endorse or promote products derived from
+       this software without specific prior written permission.
+ .
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+
+Files: JavaScriptCore/kjs/dtoa.cpp
+Copyright: © 1991, 2000, 2001 Lucent Technologies
+License: other
+ Permission to use, copy, modify, and distribute this software for any
+ purpose without fee is hereby granted, provided that this entire notice
+ is included in all copies of any software which is or includes a copy
+ or modification of this software and in all copies of the supporting
+ documentation for such software.
+ .
+ THIS SOFTWARE IS BEING PROVIDED "AS IS", WITHOUT ANY EXPRESS OR IMPLIED
+ WARRANTY.  IN PARTICULAR, NEITHER THE AUTHOR NOR LUCENT MAKES ANY
+ REPRESENTATION OR WARRANTY OF ANY KIND CONCERNING THE MERCHANTABILITY
+ OF THIS SOFTWARE OR ITS FITNESS FOR ANY PARTICULAR PURPOSE.
+
+Files: JavaScriptCore/wtf/CrossThreadRefCounted.*,
+       JavaScriptCore/wtf/FastMalloc.cpp,
+       JavaScriptCore/wtf/TC*,
+       JavaScriptCore/wtf/PtrAndFlags.h,
+       WebCore/bindings/js/InspectorBackend.*,
+       WebCore/bindings/js/JSAbstractWorkerCustom.cpp,
+       WebCore/bindings/js/JSDedicatedWorkerContextCustom.cpp,
+       WebCore/bindings/js/JSInspectorBackendCustom.cpp,
+       WebCore/bindings/js/JSMessageEventCustom.cpp,
+       WebCore/bindings/js/JSMessagePortCustom.h,
+       WebCore/bindings/js/JSSharedWorkerConstructor.h,
+       WebCore/bindings/js/ScriptArray.*,
+       WebCore/bindings/js/ScriptFunctionCall.*,
+       WebCore/bindings/js/ScriptObject.*,
+       WebCore/bindings/js/ScriptObjectQuarentine.*,
+       WebCore/bindings/js/ScriptState.cpp,
+       WebCore/bindings/v8/*,
+       WebCore/css/themeChromiumWin.css,
+       WebCore/css/themeWin.css,
+       WebCore/css/themeWinQuirks.css,
+       WebCore/dom/default/PlatformMessagePortChannel.*,
+       WebCore/dom/ErrorEvent.*,
+       WebCore/dom/MessagePortChannel.*,
+       WebCore/inspector/DOMDispatchTimelineItem.*
+       WebCore/inspector/InspectorTimelineAgent.*,
+       WebCore/inspector/TimelineItem.*,
+       WebCore/inspector/front-end/Popup.js,
+       WebCore/inspector/front-end/TimelineAgent.js,
+       WebCore/platform/CrossThreadCopier.*,
+       WebCore/platform/network/soup/SocketStreamHandleSoup.cpp,
+       WebCore/platform/text/TextEncodingDetector.*,
+       WebCore/platform/text/TextEncodingDetectorICU.*,
+       WebCore/platform/text/TextEncodingDetectorNone.*,
+       WebCore/storage/SQLTransaction*,
+       WebCore/websockets/*,
+       WebCore/workers/AbstractWorker.*,
+       WebCore/workers/DedicatedWorkerContext.*,
+       WebCore/workers/SharedWorker.*,
+       WebCore/workers/GenericWorkerTask.h,
+       WebCore/workers/WorkerContextProxy.h,
+       WebCore/workers/WorkerObjectProxy.h,
+       WebCore/workers/WorkerRunLoop.cpp,
+       WebCore/workers/WorkerRunLoop.h
+Copyright: © 2005, 2006, 2007, 2008, 2009 Google Inc.
+           © 2009 Joseph Pecoraro
+License: BSD-3
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions are
+ met:
+ .
+     * Redistributions of source code must retain the above copyright
+ notice, this list of conditions and the following disclaimer.
+     * Redistributions in binary form must reproduce the above
+ copyright notice, this list of conditions and the following disclaimer
+ in the documentation and/or other materials provided with the
+ distribution.
+     * Neither the name of Google Inc. nor the names of its
+ contributors may be used to endorse or promote products derived from
+ this software without specific prior written permission.
+ .
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+Files: JavaScriptCore/icu/*,
+       WebCore/icu/*,
+Copyright: © 1995-2006 International Business Machines Corporation and others
+License: other
+ Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ software and associated documentation files (the "Software"), to deal in the Software
+ without restriction, including without limitation the rights to use, copy, modify,
+ merge, publish, distribute, and/or sell copies of the Software, and to permit persons
+ to whom the Software is furnished to do so, provided that the above copyright notice(s)
+ and this permission notice appear in all copies of the Software and that both the above
+ copyright notice(s) and this permission notice appear in supporting documentation.
+ .
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ PURPOSE AND NONINFRINGEMENT OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER
+ OR HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL INDIRECT OR
+ CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR
+ PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
+ OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ .
+ Except as contained in this notice, the name of a copyright holder shall not be used in
+ advertising or otherwise to promote the sale, use or other dealings in this Software
+ without prior written authorization of the copyright holder.
+
+Files: WebCore/platform/image-decoders/zlib/*
+Copyright: © 1995-2005 Mark Adler
+           © 1995-2005 Jean-loup Gailly
+License: other
+ This software is provided 'as-is', without any express or implied
+ warranty.  In no event will the authors be held liable for any damages
+ arising from the use of this software.
+ .
+ Permission is granted to anyone to use this software for any purpose,
+ including commercial applications, and to alter it and redistribute it
+ freely, subject to the following restrictions:
+ .
+ 1. The origin of this software must not be misrepresented; you must not
+    claim that you wrote the original software. If you use this software
+    in a product, an acknowledgment in the product documentation would be
+    appreciated but is not required.
+ 2. Altered source versions must be plainly marked as such, and must not be
+    misrepresented as being the original software.
+ 3. This notice may not be removed or altered from any source distribution.
+
+Files: acinclude.m4
+Copyright: © 2007, 2008 Josh Triplett <josh@freedesktop.org>
+License: other
+ Copying and distribution of this file, with or without modification,
+ are permitted in any medium without royalty provided the copyright
+ notice and this notice are preserved.
+
+Files: WebCore/xml/XSLTExtensions.cpp
+Copyright: © 2001-2002 Thomas Broyer
+           © 2001-2002 Charlie Bozeman
+           © 2001-2002 Daniel Veillard
+           © 2007 Alexey Proskuryakov <ap@webkit.org>
+License: other
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is fur-
+ nished to do so, subject to the following conditions:
+ .
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+ .
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FIT-
+ NESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CON-
+ NECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ .
+ Except as contained in this notice, the name of the authors shall not
+ be used in advertising or otherwise to promote the sale, use or other deal-
+ ings in this Software without prior written authorization from him.
+
+Files: WebCore/bindings/js/JSEventSourceConstructor.*,
+       WebCore/bindings/js/JSEventSourceCustom.*,
+       WebCore/page/EventSource.*
+Copyright: © 2009 Ericsson AB
+License: BSD-3
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions
+ are met:
+
+ 1. Redistributions of source code must retain the above copyright
+    notice, this list of conditions and the following disclaimer.
+ 2. Redistributions in binary form must reproduce the above copyright
+    notice, this list of conditions and the following disclaimer
+    in the documentation and/or other materials provided with the
+    distribution.
+ 3. Neither the name of Ericsson nor the names of its contributors
+    may be used to endorse or promote products derived from this
+    software without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
diff --git a/debian/libwebkitnode-1.0-0.install b/debian/libwebkitnode-1.0-0.install
new file mode 100644
index 0000000..279ccd6
--- /dev/null
+++ b/debian/libwebkitnode-1.0-0.install
@@ -0,0 +1 @@
+usr/lib/libwebkitnode-1.0.so.*
diff --git a/debian/libwebkitnode-dev.install b/debian/libwebkitnode-dev.install
new file mode 100644
index 0000000..c50a91a
--- /dev/null
+++ b/debian/libwebkitnode-dev.install
@@ -0,0 +1,3 @@
+usr/lib/libwebkitnode-1.0.so
+usr/include/webkitnode-1.0/webkit
+usr/lib/pkgconfig/webkitnode-1.0.pc
diff --git a/debian/patches/series b/debian/patches/series
new file mode 100644
index 0000000..e69de29
diff --git a/debian/rules b/debian/rules
new file mode 100755
index 0000000..3e6466f
--- /dev/null
+++ b/debian/rules
@@ -0,0 +1,160 @@
+#!/usr/bin/make -f
+
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+DEB_HOST_GNU_TYPE   ?= $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
+DEB_BUILD_GNU_TYPE  ?= $(shell dpkg-architecture -qDEB_BUILD_GNU_TYPE)
+
+GTK_VERSIONS = 2.0
+
+CFLAGS = -g -Wall -Wl,--as-needed
+
+ifneq (,$(filter noopt,$(DEB_BUILD_OPTIONS)))
+        CFLAGS += -O0
+else
+ifeq (armhf,$(filter $(DEB_BUILD_ARCH),armhf))
+	CFLAGS += -O1
+else
+        CFLAGS += -O2
+endif
+endif
+
+ifeq ($(DEB_BUILD_ARCH),alpha)
+        CFLAGS += -Wl,--no-relax
+endif
+
+ifneq (,$(findstring $(DEB_BUILD_ARCH),mips mipsel s390 s390x))
+        CFLAGS += -gstabs
+endif
+
+ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+	NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+	MAKEARGUMENTS += -j$(NUMJOBS)
+endif
+
+clean:
+	dh_testdir
+	dh_testroot
+
+	rm -f build-stamp install-stamp
+	-test -d build/Source/WebKit/gtk/docs && \
+		cd build/Source/WebKit/gtk/docs && \
+		make maintainer-clean
+
+	-rm -rf Source/WebKit/gtk/docs/xml* Source/WebKit/gtk/docs/html* Source/WebKit/gtk/docs/sgml.stamp Source/WebKit/gtk/docs/tmpl*
+	-for file in Source/WebKit/gtk/docs/webkitgtk.prerequisites Source/WebKit/gtk/docs/tmpl.stamp \
+		Source/WebKit/gtk/docs/webkitgtk-overrides.txt Source/WebKit/gtk/docs/webkitgtk.args \
+		Source/WebKit/gtk/docs/webkitgtk.interfaces Source/WebKit/gtk/docs/webkitgtk.signals \
+		Source/WebKit/gtk/docs/webkitgtk.hierarchy Source/WebKit/gtk/docs/tmpl/webkitgtk-unused.sgml \
+		Source/WebKit/gtk/docs/webkitgtk-decl-list.txt Source/WebKit/gtk/docs/webkitgtk-decl.txt \
+		Source/WebKit/gtk/docs/webkitgtk-undeclared.txt Source/WebKit/gtk/docs/webkitgtk-undocumented.txt \
+		Source/WebKit/gtk/docs/webkitgtk-unused.txt; do \
+		rm -f ${file}; \
+	done
+	rm -rf build-*
+	rm -rf debian/tmp-*
+	rm -f Source/WebKit/gtk/docs/version.xml
+
+	for file in config.guess config.sub; do \
+		sed -i '2!b;/^exec "/ d' Source/autotools/$$file ; \
+	done
+
+	dh_clean
+
+build: build-stamp
+
+build-stamp:
+	dh_testdir
+
+	for file in config.guess config.sub; do \
+		sed -i '2!b;/^#/ i\exec "/usr/share/misc/'$$file'" "$$@"' Source/autotools/$$file ; \
+	done
+
+	# FIXME: upstream is broken when running pot-update, is why we
+	# create the po directory manually here
+	for version in $(GTK_VERSIONS); do \
+		if [ ! -d build-$${version} ]; then \
+			mkdir build-$${version}; \
+			mkdir -p build-$${version}/Source/WebKit/gtk/po/; \
+		fi; \
+		cd build-$${version}; \
+		env AR_FLAGS="cruT" \
+		CFLAGS="$(CFLAGS)" \
+		CXXFLAGS="$(CFLAGS)" \
+		CC="gcc" \
+		CXX="g++" \
+		../configure --prefix=/usr \
+			--disable-silent-rules \
+			--host=$(DEB_HOST_GNU_TYPE) \
+			--build=$(DEB_BUILD_GNU_TYPE) \
+			--with-gtk=$${version} \
+            --with-jsengine=v8 \
+		cd ..; \
+	done
+
+	for version in $(GTK_VERSIONS); do \
+		$(MAKE) $(MAKEARGUMENTS) -C build-$${version}; \
+	done
+
+	touch $@
+
+install: install-stamp
+
+install-stamp: build-stamp
+	dh_testdir
+	dh_testroot
+
+	dh_clean -k
+
+ifeq (2.0,$(filter 2.0,$(GTK_VERSIONS)))
+	$(MAKE) -C build-2.0 install DESTDIR="$(CURDIR)"/debian/tmp-2.0
+	[ ! -d debian/tmp-2.0/usr/lib/webkitgtk-1.0-0/libexec ] && install -d -m 755 debian/tmp-2.0/usr/lib/webkitgtk-1.0-0/libexec || true
+
+endif
+
+	touch $@
+
+# Build architecture-independent files here.
+binary-indep: build install
+	dh_testdir
+	dh_testroot
+	dh_installdocs -i
+	dh_installchangelogs -i
+	dh_link -i
+	dh_compress -i
+	dh_fixperms -i
+	dh_installdeb -i
+	dh_gencontrol -i
+	dh_md5sums -i
+	dh_builddeb -i
+
+# Build architecture-dependent files here.
+binary-arch: build install
+	dh_testdir
+	dh_testroot
+	dh_installdocs -a
+	dh_installchangelogs -a
+ifeq (2.0,$(filter 2.0,$(GTK_VERSIONS)))
+	dh_install -plibwebkitnode-1.0-0 --sourcedir=debian/tmp-2.0
+	dh_install -plibwebkitnode-dev --sourcedir=debian/tmp-2.0
+	dh_install -plibwebkitnode-1.0-0-dbg --sourcedir=debian/tmp-2.0
+endif
+	dh_link -a
+ifeq (2.0,$(filter 2.0,$(GTK_VERSIONS)))
+	dh_strip -plibwebkitnode-1.0-0 --dbg-package=libwebkitnode-1.0-0-dbg
+endif
+	dh_compress -a
+	dh_fixperms -a
+ifeq (2.0,$(filter 2.0,$(GTK_VERSIONS)))
+	dh_makeshlibs -plibwebkitnode-1.0-0 -V 'libwebkitnode-1.0-0 (>= 1.3.9)' -- -c4
+endif
+	dh_installdeb -a
+	dh_shlibdeps -a
+	dh_gencontrol -a
+	dh_md5sums -a
+	dh_builddeb -a
+
+binary: binary-indep binary-arch
+
+.PHONY: build clean install binary binary-indep binary-arch
diff --git a/debian/source/format b/debian/source/format
new file mode 100644
index 0000000..163aaf8
--- /dev/null
+++ b/debian/source/format
@@ -0,0 +1 @@
+3.0 (quilt)
diff --git a/debian/source/lintian-overrides b/debian/source/lintian-overrides
new file mode 100644
index 0000000..ef8c395
--- /dev/null
+++ b/debian/source/lintian-overrides
@@ -0,0 +1 @@
+webkit source: build-depends-on-1-revision build-depends: libgtk2.0-dev (>= 2.24.3-1)
diff --git a/debian/watch b/debian/watch
new file mode 100644
index 0000000..2427378
--- /dev/null
+++ b/debian/watch
@@ -0,0 +1,2 @@
+version=3
+http://webkitgtk.org/?page=download webkit-([\d]+\.[\d]*[02468]\.[\d]+).tar.gz
-- 
1.7.8.2

